{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "116d49e2",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import glob\n",
    "import subprocess as sp\n",
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "26ab494f",
   "metadata": {},
   "outputs": [],
   "source": [
    "root = os.path.join(\"/data/manke/group/rabbani/atac_project2078/\")\n",
    "bw_path = os.path.join(root, \"atac_pipeline_merged_bam\", \"BigWigs\")\n",
    "rna_path = os.path.join(\"/data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros\")\n",
    "output_path = os.path.join(\"/data/akhtar/group2/rabbani/atac_project2078/plots\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "ad09b4c8",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/localenv/rabbani/anaconda/miniconda3/envs/matplotlib/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3444: DtypeWarning: Columns (0) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  exec(code_obj, self.user_global_ns, self.user_ns)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                  GeneID     baseMean  log2FoldChange     lfcSE      stat  \\\n",
      "0   ENSMUSG00000108365.1    11.316024        1.944718  0.871289  2.232002   \n",
      "1  ENSMUSG00000033632.15   437.496242       -0.001246  0.083726 -0.014887   \n",
      "2   ENSMUSG00000078314.2     7.868302       -0.033810  0.573842 -0.058918   \n",
      "3  ENSMUSG00000021668.14  1454.583425        0.047710  0.062908  0.758413   \n",
      "4  ENSMUSG00000022329.14  2422.361965       -0.024019  0.043861 -0.547622   \n",
      "\n",
      "     pvalue  padj Status external_gene_name   0         3         4  \\\n",
      "0  0.025615   NaN   None       RP24-174G2.2   7  80036705  80038728   \n",
      "1  0.988122   NaN   None           AW554918  18  25168999  25467321   \n",
      "2  0.953017   NaN   None            Gm14762   X  85825892  85834831   \n",
      "3  0.448204   NaN   None               Polk  13  96480689  96542485   \n",
      "4  0.583951   NaN   None               Stk3  15  34875496  35155806   \n",
      "\n",
      "                       8  5  6  \n",
      "0   ENSMUSG00000108365.1  .  +  \n",
      "1  ENSMUSG00000033632.15  .  +  \n",
      "2   ENSMUSG00000078314.2  .  +  \n",
      "3  ENSMUSG00000021668.14  .  -  \n",
      "4  ENSMUSG00000022329.14  .  -  \n"
     ]
    }
   ],
   "source": [
    "# make bed file\n",
    "gene_coord = os.path.join(\"/data/manke/group/rabbani/rna_project2129/genes.filtered.gtf\")\n",
    "gene_coord = pd.read_csv(gene_coord, sep = \"\\t\", header = None, comment = \"#\")\n",
    "gene_coord = gene_coord.loc[gene_coord[2] == \"gene\"]\n",
    "gene_coord[8] = gene_coord[8].str.split('\"', 2, expand=True)[1]\n",
    "gene_coord = gene_coord[[0,3,4,8,5,6]]\n",
    "os.makedirs(os.path.join(rna_path, \"beds\"), exist_ok= True)\n",
    "# Make UP and DOWN bed files per time point\n",
    "for time in ['LPS0', 'LPS3', 'LPS12']:\n",
    "    for index, cond in enumerate([\"vector\", \"K197R\", \"K197Q\"]):\n",
    "        files = glob.glob(os.path.join(rna_path, \"_\".join([cond,time,\"vs\", \"WT\",time])+\".tsv\"))\n",
    "        assert(len(files) == 1)\n",
    "        file = files[0]\n",
    "        name = os.path.basename(file).split(\".tsv\")[0]\n",
    "        df = pd.read_csv(file, sep = \"\\t\")\n",
    "        bed = df.merge(gene_coord, left_on = \"GeneID\", right_on = gene_coord[8], how = \"inner\")\n",
    "        up = bed.loc[bed[\"Status\"] == \"UP\"]\n",
    "        down = bed.loc[bed[\"Status\"] == \"DOWN\"]\n",
    "        if len(down) > 0:\n",
    "            down = down[[0, 3, 4, 8, 5, 6]]\n",
    "            down.columns = ['#chr', 'start', 'end', 'GeneID', 'score', 'strand']\n",
    "            down.to_csv(os.path.join(rna_path, \"beds\", name + \"_down.bed\"), sep = \"\\t\", index = False)\n",
    "        if len(up) > 0:\n",
    "            up = up[[0, 3, 4, 8, 5, 6]]\n",
    "            up.columns = ['#chr', 'start', 'end', 'GeneID', 'score', 'strand']\n",
    "            up.to_csv(os.path.join(rna_path, \"beds\", name + \"_up.bed\"), sep = \"\\t\", index = False)\n",
    "#  Make 500 random genes bed\n",
    "random_genes = pd.read_csv(os.path.join(rna_path, \"K197R_LPS12_vs_WT_LPS12.tsv\"), sep = \"\\t\")\n",
    "random_genes = random_genes.sample(n = 500)\n",
    "random_genes = random_genes.merge(gene_coord, left_on = \"GeneID\", right_on = gene_coord[8], how = \"inner\")\n",
    "print(random_genes.head())\n",
    "random_genes = random_genes[[0, 3, 4, 8, 5, 6]]\n",
    "random_genes.columns = ['#chr', 'start', 'end', 'GeneID', 'score', 'strand']\n",
    "random_genes.to_csv(os.path.join(rna_path, \"beds\", \"500RandomGenes.bed\"), sep = \"\\t\", index = False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "20078e24",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "K197Q_LPS12_vs_WT_LPS12_down.bed 807 764 754\n",
      "vector_LPS0_vs_WT_LPS0_down.bed 709 702 701\n",
      "K197Q_LPS0_vs_WT_LPS0_down.bed 756 746 735\n",
      "K197R_LPS0_vs_WT_LPS0_down.bed 525 524 520\n",
      "vector_LPS12_vs_WT_LPS12_down.bed 471 459 456\n",
      "vector_LPS3_vs_WT_LPS3_up.bed 624 612 611\n",
      "K197Q_LPS12_vs_WT_LPS12_up.bed 1768 1734 1730\n",
      "K197Q_LPS3_vs_WT_LPS3_down.bed 611 618 607\n",
      "K197R_LPS0_vs_WT_LPS0_up.bed 375 375 372\n",
      "K197R_LPS3_vs_WT_LPS3_down.bed 195 194 192\n",
      "vector_LPS3_vs_WT_LPS3_down.bed 624 611 607\n",
      "vector_LPS0_vs_WT_LPS0_up.bed 289 269 267\n",
      "vector_LPS12_vs_WT_LPS12_up.bed 504 480 478\n",
      "this is only found after removing zeros! 500RandomGenes.bed 500\n",
      "K197R_LPS3_vs_WT_LPS3_up.bed 199 197 197\n",
      "K197Q_LPS0_vs_WT_LPS0_up.bed 641 610 606\n",
      "K197Q_LPS3_vs_WT_LPS3_up.bed 358 356 354\n"
     ]
    }
   ],
   "source": [
    "# compare wo_zeros up/down with vs_wt up/down. This is just to see how much they are in agreement\n",
    "for file in glob.glob(os.path.join(rna_path, \"beds\", \"*.bed\")):\n",
    "    name = os.path.basename(file)\n",
    "    df = pd.read_csv(file, sep = \"\\t\")\n",
    "    try:\n",
    "        df1 = pd.read_csv(os.path.join(\"/data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_vs_wt/beds/\", name), sep = \"\\t\", header = None)\n",
    "        df1.columns = ['#chr', 'start', 'end']\n",
    "        merged_df = df.merge(df1, on = ['#chr', 'start', 'end'], how = \"inner\")\n",
    "        print(name, len(df), len(df1), len(merged_df))\n",
    "    except:\n",
    "        print(\"this is only found after removing zeros!\", name, len(df))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "871b3c84",
   "metadata": {},
   "outputs": [],
   "source": [
    "# computematrix on diff tss +-2kb\n",
    "# plot heatmap\n",
    "# profile plot\n",
    "# ultraheatmap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "928fe56d",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools;computeMatrix reference-point -S /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_WT_LPS0.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_Vector_LPS0.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_K197R_LPS0.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_K197Q_LPS0.bw  -R /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/beds/K197R_LPS0_vs_WT_LPS0_down.bed /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/beds/K197R_LPS0_vs_WT_LPS0_up.bed /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/beds/K197Q_LPS0_vs_WT_LPS0_down.bed /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/beds/K197Q_LPS0_vs_WT_LPS0_up.bed /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/beds/vector_LPS0_vs_WT_LPS0_down.bed /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/beds/vector_LPS0_vs_WT_LPS0_up.bed  -a 2000 -b 2000  --samplesLabel WT_LPS0 Vector_LPS0 K197R_LPS0 K197Q_LPS0  -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS0.gz; plotHeatmap -m /data/akhtar/group2/rabbani/atac_project2078/plots/LPS0.gz -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS0.png; plotProfile -m /data/akhtar/group2/rabbani/atac_project2078/plots/LPS0.gz --perGroup -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS0_profile.png --outFileNameData /data/akhtar/group2/rabbani/atac_project2078/plots/LPS0_profile.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper.py:1219: RuntimeWarning:\n",
      "\n",
      "Mean of empty slice\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"RdYlBu\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:762: MatplotlibDeprecationWarning:\n",
      "\n",
      "The 'alpha' parameter to Colorbar has no effect because it is overridden by the mappable; it is deprecated since 3.3 and will be removed two minor releases later.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:770: MatplotlibDeprecationWarning:\n",
      "\n",
      "savefig() got unexpected keyword argument \"pdd_inches\" which is no longer supported as of 3.3 and will become an error two minor releases later\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools;computeMatrix reference-point -S /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_WT_LPS3.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_Vector_LPS3.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_K197R_LPS3.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_K197Q_LPS3.bw  -R /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/beds/K197R_LPS3_vs_WT_LPS3_down.bed /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/beds/K197R_LPS3_vs_WT_LPS3_up.bed /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/beds/K197Q_LPS3_vs_WT_LPS3_down.bed /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/beds/K197Q_LPS3_vs_WT_LPS3_up.bed /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/beds/vector_LPS3_vs_WT_LPS3_down.bed /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/beds/vector_LPS3_vs_WT_LPS3_up.bed  -a 2000 -b 2000  --samplesLabel WT_LPS3 Vector_LPS3 K197R_LPS3 K197Q_LPS3  -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS3.gz; plotHeatmap -m /data/akhtar/group2/rabbani/atac_project2078/plots/LPS3.gz -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS3.png; plotProfile -m /data/akhtar/group2/rabbani/atac_project2078/plots/LPS3.gz --perGroup -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS3_profile.png --outFileNameData /data/akhtar/group2/rabbani/atac_project2078/plots/LPS3_profile.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper.py:1219: RuntimeWarning:\n",
      "\n",
      "Mean of empty slice\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"RdYlBu\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:762: MatplotlibDeprecationWarning:\n",
      "\n",
      "The 'alpha' parameter to Colorbar has no effect because it is overridden by the mappable; it is deprecated since 3.3 and will be removed two minor releases later.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:770: MatplotlibDeprecationWarning:\n",
      "\n",
      "savefig() got unexpected keyword argument \"pdd_inches\" which is no longer supported as of 3.3 and will become an error two minor releases later\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools;computeMatrix reference-point -S /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_WT_LPS12.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_Vector_LPS12.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_K197R_LPS12.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_K197Q_LPS12.bw  -R /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/beds/K197Q_LPS12_vs_WT_LPS12_down.bed /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/beds/K197Q_LPS12_vs_WT_LPS12_up.bed /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/beds/vector_LPS12_vs_WT_LPS12_down.bed /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/beds/vector_LPS12_vs_WT_LPS12_up.bed /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/beds/500RandomGenes.bed  -a 2000 -b 2000  --samplesLabel WT_LPS12 Vector_LPS12 K197R_LPS12 K197Q_LPS12  -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS12.gz; plotHeatmap -m /data/akhtar/group2/rabbani/atac_project2078/plots/LPS12.gz -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS12.png; plotProfile -m /data/akhtar/group2/rabbani/atac_project2078/plots/LPS12.gz --perGroup -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS12_profile.png --outFileNameData /data/akhtar/group2/rabbani/atac_project2078/plots/LPS12_profile.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper.py:1219: RuntimeWarning:\n",
      "\n",
      "Mean of empty slice\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"RdYlBu\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:762: MatplotlibDeprecationWarning:\n",
      "\n",
      "The 'alpha' parameter to Colorbar has no effect because it is overridden by the mappable; it is deprecated since 3.3 and will be removed two minor releases later.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:770: MatplotlibDeprecationWarning:\n",
      "\n",
      "savefig() got unexpected keyword argument \"pdd_inches\" which is no longer supported as of 3.3 and will become an error two minor releases later\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for time in ['LPS0', 'LPS3', 'LPS12']:\n",
    "    bws = \"\"\n",
    "    names = \"\"\n",
    "    rna = \"\"\n",
    "    \n",
    "    for index, cond in enumerate([\"WT\",\"Vector\", \"K197R\", \"K197Q\"]):\n",
    "        bws += os.path.join(bw_path, \"_\".join([\"merged\", cond,time])+\".bw\")+\" \"\n",
    "        names += \"_\".join([cond,time])+ \" \"\n",
    "    for index, cond in enumerate([\"K197R\", \"K197Q\", \"vector\"]):\n",
    "        down_file = os.path.join(rna_path, \"beds\", \"_\".join([cond,time,\"vs\", \"WT\",time])+\"_down.bed\")\n",
    "        \n",
    "        if  os.path.isfile(down_file):\n",
    "            rna += down_file + \" \"\n",
    "        up_file = os.path.join(rna_path, \"beds\", \"_\".join([cond,time,\"vs\", \"WT\",time])+\"_up.bed\")\n",
    "        if os.path.isfile(up_file):\n",
    "            rna += up_file + \" \"\n",
    "    ##Add 500 random genes\n",
    "    if time == 'LPS12':\n",
    "        rand = os.path.join(rna_path, \"beds\", \"500RandomGenes.bed\")\n",
    "        rna += rand + \" \"\n",
    "    if rna != \"\":\n",
    "        cmd = \"module load deeptools/3.5.0;\"\n",
    "        cmd += \"computeMatrix reference-point -S \"+bws\n",
    "        cmd += \" -R \"+rna\n",
    "        cmd += \" -a 2000 -b 2000 \"\n",
    "        cmd += \" --samplesLabel \"+ names\n",
    "        cmd += \" -o \"+os.path.join(output_path, time+\".gz\")\n",
    "        cmd += \"; plotHeatmap -m \"+os.path.join(output_path, time+\".gz\")\n",
    "        cmd += \" -o \"+os.path.join(output_path, time+\".png\")\n",
    "        cmd += \"; plotProfile -m \"+os.path.join(output_path, time+\".gz\")\n",
    "        cmd += \" --perGroup -o \"+os.path.join(output_path, time+\"_profile.png\")\n",
    "        cmd += \" --outFileNameData \"+os.path.join(output_path, time+\"_profile.tsv\")\n",
    "        print(cmd)\n",
    "        sp.check_output(cmd, shell = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "cc9c9ffe",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load ultraheatmap; addFeatureToMatrix -m /data/akhtar/group2/rabbani/atac_project2078/plots/LPS0.gz -o /data/akhtar/group2/rabbani/atac_project2078/plots/ultraheatmap/LPS0_w_deg.gz -t /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/K197Q_LPS0_vs_WT_LPS0.tsv /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/K197R_LPS0_vs_WT_LPS0.tsv /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/vector_LPS0_vs_WT_LPS0.tsv --featureIdColumn 'GeneID' --featureNames 'log2FoldChange' \n",
      "module load ultraheatmap; addFeatureToMatrix -m /data/akhtar/group2/rabbani/atac_project2078/plots/LPS3.gz -o /data/akhtar/group2/rabbani/atac_project2078/plots/ultraheatmap/LPS3_w_deg.gz -t /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/K197Q_LPS3_vs_WT_LPS3.tsv /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/K197R_LPS3_vs_WT_LPS3.tsv /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/vector_LPS3_vs_WT_LPS3.tsv --featureIdColumn 'GeneID' --featureNames 'log2FoldChange' \n",
      "module load ultraheatmap; addFeatureToMatrix -m /data/akhtar/group2/rabbani/atac_project2078/plots/LPS12.gz -o /data/akhtar/group2/rabbani/atac_project2078/plots/ultraheatmap/LPS12_w_deg.gz -t /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/K197Q_LPS12_vs_WT_LPS12.tsv /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/K197R_LPS12_vs_WT_LPS12.tsv /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/vector_LPS12_vs_WT_LPS12.tsv --featureIdColumn 'GeneID' --featureNames 'log2FoldChange' \n"
     ]
    }
   ],
   "source": [
    "# ultraheatmap wo mapping on the above matrix \n",
    "for time in ['LPS0', 'LPS3', 'LPS12']:\n",
    "    cmd = \"module load ultraheatmap/1.3.1; addFeatureToMatrix \"\n",
    "    cmd += \"-m \"+os.path.join(output_path, time+\".gz\")\n",
    "    cmd += \" -o \"+os.path.join(output_path, \"ultraheatmap\", time+\"_w_deg.gz\")\n",
    "    cmd += \" -t \"+os.path.join(rna_path, \"K197Q_\"+time+\"_vs_WT_\"+time+\".tsv\")+ \" \"+os.path.join(rna_path, \"K197R_\"+time+\"_vs_WT_\"+time+\".tsv\")\n",
    "    cmd += \" \"+os.path.join(rna_path, \"vector_\"+time+\"_vs_WT_\"+time+\".tsv\")\n",
    "    cmd += \" --featureIdColumn 'GeneID' --featureNames 'log2FoldChange' \"\n",
    "    print(cmd)\n",
    "    sp.check_output(cmd, shell = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "50ef2844",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools; plotHeatmap  -m /data/akhtar/group2/rabbani/atac_project2078/plots/ultraheatmap/LPS0_w_deg.gz -o /data/akhtar/group2/rabbani/atac_project2078/plots/ultraheatmap/LPS0_w_deg.png --zMin 0 0 0 0 -2 -2 -2 --zMax 1 1 1 1 2 2 2  --colorMap Greens Greens Greens Greens RdBu_r RdBu_r RdBu_r --sortUsingSample 5 6 7 --samplesLabel wt_atac vector_atac kr_atac kq_atac kq_vs_wt kr_vs_wt vector_vs_wt \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"Greens\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"RdBu_r\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:738: MatplotlibDeprecationWarning:\n",
      "\n",
      "The 'alpha' parameter to Colorbar has no effect because it is overridden by the mappable; it is deprecated since 3.3 and will be removed two minor releases later.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:770: MatplotlibDeprecationWarning:\n",
      "\n",
      "savefig() got unexpected keyword argument \"pdd_inches\" which is no longer supported as of 3.3 and will become an error two minor releases later\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools; plotHeatmap  -m /data/akhtar/group2/rabbani/atac_project2078/plots/ultraheatmap/LPS3_w_deg.gz -o /data/akhtar/group2/rabbani/atac_project2078/plots/ultraheatmap/LPS3_w_deg.png --zMin 0 0 0 0 -2 -2 -2 --zMax 1 1 1 1 2 2 2  --colorMap Greens Greens Greens Greens RdBu_r RdBu_r RdBu_r --sortUsingSample 5 6 7 --samplesLabel wt_atac vector_atac kr_atac kq_atac kq_vs_wt kr_vs_wt vector_vs_wt \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"Greens\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"RdBu_r\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:738: MatplotlibDeprecationWarning:\n",
      "\n",
      "The 'alpha' parameter to Colorbar has no effect because it is overridden by the mappable; it is deprecated since 3.3 and will be removed two minor releases later.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:770: MatplotlibDeprecationWarning:\n",
      "\n",
      "savefig() got unexpected keyword argument \"pdd_inches\" which is no longer supported as of 3.3 and will become an error two minor releases later\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools; plotHeatmap  -m /data/akhtar/group2/rabbani/atac_project2078/plots/ultraheatmap/LPS12_w_deg.gz -o /data/akhtar/group2/rabbani/atac_project2078/plots/ultraheatmap/LPS12_w_deg.png --zMin 0 0 0 0 -2 -2 -2 --zMax 1 1 1 1 2 2 2  --colorMap Greens Greens Greens Greens RdBu_r RdBu_r RdBu_r --sortUsingSample 5 6 7 --samplesLabel wt_atac vector_atac kr_atac kq_atac kq_vs_wt kr_vs_wt vector_vs_wt \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"Greens\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"RdBu_r\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:738: MatplotlibDeprecationWarning:\n",
      "\n",
      "The 'alpha' parameter to Colorbar has no effect because it is overridden by the mappable; it is deprecated since 3.3 and will be removed two minor releases later.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:770: MatplotlibDeprecationWarning:\n",
      "\n",
      "savefig() got unexpected keyword argument \"pdd_inches\" which is no longer supported as of 3.3 and will become an error two minor releases later\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# plotheatmap\n",
    "for time in ['LPS0', 'LPS3', 'LPS12']:\n",
    "    cmd = \"module load deeptools/3.5.0; plotHeatmap \"\n",
    "    cmd += \" -m \"+os.path.join(output_path, \"ultraheatmap\", time+\"_w_deg.gz\")\n",
    "    cmd += \" -o \"+os.path.join(output_path, \"ultraheatmap\", time+\"_w_deg.png\")\n",
    "    cmd += \" --zMin 0 0 0 0 -2 -2 -2 --zMax 1 1 1 1 2 2 2 \" \n",
    "    cmd += \" --colorMap Greens Greens Greens Greens RdBu_r RdBu_r RdBu_r --sortUsingSample 5 6 7 --samplesLabel wt_atac vector_atac kr_atac kq_atac kq_vs_wt kr_vs_wt vector_vs_wt \"\n",
    "    print(cmd)\n",
    "    sp.check_output(cmd, shell = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "77411a89",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ultraheatmap\n",
    "# 1. plot bws on peaks (ocrs)\n",
    "# 2. add gene expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "0f5ab2a7",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools/3.5.0;computeMatrix reference-point -S /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_Vector_LPS0.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_WT_LPS0.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_K197R_LPS0.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_K197Q_LPS0.bw  -R /data/akhtar/group2/rabbani/atac_project2078/hmmratac/ocrs/merged_peaks/merged_Vector_LPS0_.bed /data/akhtar/group2/rabbani/atac_project2078/hmmratac/ocrs/merged_peaks/merged_WT_LPS0_.bed /data/akhtar/group2/rabbani/atac_project2078/hmmratac/ocrs/merged_peaks/merged_K197R_LPS0_.bed /data/akhtar/group2/rabbani/atac_project2078/hmmratac/ocrs/merged_peaks/merged_K197Q_LPS0_.bed  -a 2000 -b 2000 --referencePoint center -p 20  --missingDataAsZero  --skipZeros  --samplesLabel Vector_LPS0 WT_LPS0 K197R_LPS0 K197Q_LPS0  --blackListFileName /data/repository/organisms/GRCm38_ensembl/akundaje/blacklist.bed  -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS0_onpeaks.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Skipping 1:88224080-88224340, due to being absent in the computeMatrix output.\n",
      "Skipping 1:183299130-183299590, due to being absent in the computeMatrix output.\n",
      "Skipping 11:109011640-109012030, due to being absent in the computeMatrix output.\n",
      "Skipping 13:119613130-119613380, due to being absent in the computeMatrix output.\n",
      "Skipping 13:119614490-119614720, due to being absent in the computeMatrix output.\n",
      "Skipping 13:119616350-119616730, due to being absent in the computeMatrix output.\n",
      "Skipping 13:119617260-119617490, due to being absent in the computeMatrix output.\n",
      "Skipping 15:75086260-75086530, due to being absent in the computeMatrix output.\n",
      "Skipping 15:75086680-75087040, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39842990-39843480, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39844792-39845282, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39845519-39845749, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39847720-39848840, due to being absent in the computeMatrix output.\n",
      "Skipping 18:68692070-68692280, due to being absent in the computeMatrix output.\n",
      "Skipping 2:98662240-98662900, due to being absent in the computeMatrix output.\n",
      "Skipping 5:146260990-146261350, due to being absent in the computeMatrix output.\n",
      "Skipping GL456210.1:125630-125930, due to being absent in the computeMatrix output.\n",
      "Skipping GL456210.1:141970-142410, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:36830-37100, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:56300-56720, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:139550-140100, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:123880-124650, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:125490-125940, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:150460-150670, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:153070-153300, due to being absent in the computeMatrix output.\n",
      "Skipping GL456216.1:13760-14030, due to being absent in the computeMatrix output.\n",
      "Skipping GL456216.1:15660-17500, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:36390-36690, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:64600-65500, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:85810-86270, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:38730-39500, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:138950-139310, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:159520-160150, due to being absent in the computeMatrix output.\n",
      "Skipping GL456239.1:37790-38520, due to being absent in the computeMatrix output.\n",
      "Skipping GL456359.1:19330-19870, due to being absent in the computeMatrix output.\n",
      "Skipping JH584295.1:1620-1960, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:280-530, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:680-1140, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:1500-3120, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:4030-5090, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:5650-5850, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:6250-7330, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:8850-10460, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:13410-13720, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:22570-22770, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:24000-24350, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:24430-24940, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:25280-25590, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:26340-27530, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:29350-29760, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:31050-33670, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:33880-36410, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:36690-37740, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:38210-38980, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:39160-40830, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:41120-41570, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:43230-45340, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:46460-47040, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:47100-49480, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:49850-50730, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:51030-51990, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:52760-55900, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:58190-59700, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:59848-61688, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:62173-63893, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:64113-66653, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:67063-68445, due to being absent in the computeMatrix output.\n",
      "Skipping MT:6545-6975, due to being absent in the computeMatrix output.\n",
      "Skipping MT:10345-10920, due to being absent in the computeMatrix output.\n",
      "Skipping 1:88223390-88224320, due to being absent in the computeMatrix output.\n",
      "Skipping 1:183299140-183299590, due to being absent in the computeMatrix output.\n",
      "Skipping 11:109011640-109012040, due to being absent in the computeMatrix output.\n",
      "Skipping 13:119613040-119613470, due to being absent in the computeMatrix output.\n",
      "Skipping 13:119614590-119614940, due to being absent in the computeMatrix output.\n",
      "Skipping 13:119615480-119616120, due to being absent in the computeMatrix output.\n",
      "Skipping 13:119616190-119617640, due to being absent in the computeMatrix output.\n",
      "Skipping 15:75085690-75086070, due to being absent in the computeMatrix output.\n",
      "Skipping 15:75086260-75086530, due to being absent in the computeMatrix output.\n",
      "Skipping 15:75086680-75087040, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39842990-39843780, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39843795-39844135, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39845509-39845749, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39845768-39845988, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39846008-39846908, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39847720-39848840, due to being absent in the computeMatrix output.\n",
      "Skipping 18:68692070-68692290, due to being absent in the computeMatrix output.\n",
      "Skipping 2:98662240-98662900, due to being absent in the computeMatrix output.\n",
      "Skipping 5:146260990-146261360, due to being absent in the computeMatrix output.\n",
      "Skipping GL456210.1:125620-125940, due to being absent in the computeMatrix output.\n",
      "Skipping GL456210.1:141820-142460, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:36860-37090, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:56210-56730, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:139540-140150, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:123900-124650, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:125640-126040, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:150470-150670, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:153080-153320, due to being absent in the computeMatrix output.\n",
      "Skipping GL456216.1:13750-14030, due to being absent in the computeMatrix output.\n",
      "Skipping GL456216.1:15640-17480, due to being absent in the computeMatrix output.\n",
      "Skipping GL456216.1:34130-34500, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:36400-36680, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:64700-65510, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:85520-86280, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:38540-39500, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:138840-139310, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:159510-160040, due to being absent in the computeMatrix output.\n",
      "Skipping GL456239.1:1550-2010, due to being absent in the computeMatrix output.\n",
      "Skipping GL456239.1:33680-34560, due to being absent in the computeMatrix output.\n",
      "Skipping GL456239.1:34870-35420, due to being absent in the computeMatrix output.\n",
      "Skipping GL456239.1:36870-39840, due to being absent in the computeMatrix output.\n",
      "Skipping GL456359.1:19330-19780, due to being absent in the computeMatrix output.\n",
      "Skipping GL456372.1:20550-20820, due to being absent in the computeMatrix output.\n",
      "Skipping GL456387.1:11090-11420, due to being absent in the computeMatrix output.\n",
      "Skipping JH584295.1:260-560, due to being absent in the computeMatrix output.\n",
      "Skipping JH584295.1:1370-1920, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:290-510, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:670-1210, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:1500-3200, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:3990-5140, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:8840-10480, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:13430-13740, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:16140-16530, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:16890-17250, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:18830-19190, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:19280-19490, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:19820-20300, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:20800-21450, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:22530-22780, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:23990-25730, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:25930-28350, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:29110-29770, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:30980-33680, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:33880-36410, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:36660-40860, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:41010-41570, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:42860-45380, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:46470-49090, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:49220-49460, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:49550-51280, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:51360-52010, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:52590-56020, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:58210-59710, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:62160-66630, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:67055-68502, due to being absent in the computeMatrix output.\n",
      "Skipping MT:10290-10920, due to being absent in the computeMatrix output.\n",
      "Skipping 1:58613950-58615140, due to being absent in the computeMatrix output.\n",
      "Skipping 1:88223610-88224370, due to being absent in the computeMatrix output.\n",
      "Skipping 1:183299140-183299590, due to being absent in the computeMatrix output.\n",
      "Skipping 11:109011640-109012040, due to being absent in the computeMatrix output.\n",
      "Skipping 13:119613020-119613400, due to being absent in the computeMatrix output.\n",
      "Skipping 13:119616170-119617600, due to being absent in the computeMatrix output.\n",
      "Skipping 15:75085690-75086090, due to being absent in the computeMatrix output.\n",
      "Skipping 15:75086260-75086530, due to being absent in the computeMatrix output.\n",
      "Skipping 15:75086640-75087040, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39842990-39843780, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39845768-39845988, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39846006-39846906, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39847720-39848840, due to being absent in the computeMatrix output.\n",
      "Skipping 18:68692070-68692280, due to being absent in the computeMatrix output.\n",
      "Skipping 2:98662230-98662480, due to being absent in the computeMatrix output.\n",
      "Skipping 2:98662500-98662880, due to being absent in the computeMatrix output.\n",
      "Skipping 5:146260990-146261350, due to being absent in the computeMatrix output.\n",
      "Skipping GL456210.1:125640-125940, due to being absent in the computeMatrix output.\n",
      "Skipping GL456210.1:141940-142640, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:36800-37100, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:56310-56720, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:139550-140390, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:174720-174950, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:123880-124750, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:125470-125960, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:153000-153300, due to being absent in the computeMatrix output.\n",
      "Skipping GL456216.1:13760-14040, due to being absent in the computeMatrix output.\n",
      "Skipping GL456216.1:15650-17480, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:36410-36680, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:64720-65500, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:85780-86270, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:38530-39500, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:138740-139300, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:159510-160050, due to being absent in the computeMatrix output.\n",
      "Skipping GL456239.1:33560-36400, due to being absent in the computeMatrix output.\n",
      "Skipping GL456239.1:37780-39980, due to being absent in the computeMatrix output.\n",
      "Skipping GL456359.1:19320-19850, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:280-510, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:670-3170, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:4020-5420, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:5430-5850, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:5940-7350, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:9120-11020, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:12250-12460, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:13380-13730, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:15820-16020, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:16120-16530, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:16870-17280, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:19110-19490, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:19850-20280, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:20810-21380, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:22530-22770, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:23970-25040, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:25130-25660, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:25950-27670, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:27830-29020, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:29300-30040, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:30980-36400, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:36690-37920, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:38170-38990, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:39230-40840, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:40970-41490, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:43070-45340, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:46660-49460, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:49550-52120, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:57450-57740, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:58190-59710, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:62181-66834, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:67051-68454, due to being absent in the computeMatrix output.\n",
      "Skipping MT:10226-10920, due to being absent in the computeMatrix output.\n",
      "Skipping 1:58614060-58615160, due to being absent in the computeMatrix output.\n",
      "Skipping 1:88218810-88219750, due to being absent in the computeMatrix output.\n",
      "Skipping 1:88220460-88221490, due to being absent in the computeMatrix output.\n",
      "Skipping 1:183299140-183299590, due to being absent in the computeMatrix output.\n",
      "Skipping 11:109011640-109012040, due to being absent in the computeMatrix output.\n",
      "Skipping 13:119612950-119613960, due to being absent in the computeMatrix output.\n",
      "Skipping 13:119614180-119617600, due to being absent in the computeMatrix output.\n",
      "Skipping 15:75085700-75086140, due to being absent in the computeMatrix output.\n",
      "Skipping 15:75086250-75087040, due to being absent in the computeMatrix output.\n",
      "Skipping 17:36231210-36232020, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39846681-39846901, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39847720-39848840, due to being absent in the computeMatrix output.\n",
      "Skipping 18:68692070-68692290, due to being absent in the computeMatrix output.\n",
      "Skipping 2:98662240-98662480, due to being absent in the computeMatrix output.\n",
      "Skipping 2:98662510-98662880, due to being absent in the computeMatrix output.\n",
      "Skipping 5:146260990-146261360, due to being absent in the computeMatrix output.\n",
      "Skipping GL456210.1:125640-125930, due to being absent in the computeMatrix output.\n",
      "Skipping GL456210.1:141900-142650, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:56350-56760, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:116130-116440, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:139550-140170, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:123890-124890, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:125790-126300, due to being absent in the computeMatrix output.\n",
      "Skipping GL456216.1:13750-14020, due to being absent in the computeMatrix output.\n",
      "Skipping GL456216.1:15630-17480, due to being absent in the computeMatrix output.\n",
      "Skipping GL456216.1:34160-34550, due to being absent in the computeMatrix output.\n",
      "Skipping GL456216.1:34670-35050, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:36380-36680, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:64740-65500, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:85770-86370, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:38520-39500, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:138940-139300, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:159520-160050, due to being absent in the computeMatrix output.\n",
      "Skipping GL456239.1:33560-36140, due to being absent in the computeMatrix output.\n",
      "Skipping GL456239.1:36340-37600, due to being absent in the computeMatrix output.\n",
      "Skipping GL456239.1:37800-39930, due to being absent in the computeMatrix output.\n",
      "Skipping GL456359.1:19320-19560, due to being absent in the computeMatrix output.\n",
      "Skipping GL456359.1:19750-20070, due to being absent in the computeMatrix output.\n",
      "Skipping GL456366.1:28250-28560, due to being absent in the computeMatrix output.\n",
      "Skipping GL456387.1:11570-12070, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:260-510, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:680-1370, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:1490-3390, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:3540-3830, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:3960-5110, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:6220-7470, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:8870-10290, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:12240-12460, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:12690-13120, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:13380-13860, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:15800-16040, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:16120-17240, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:19270-19490, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:19830-20330, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:20800-21380, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:22530-22790, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:23990-25050, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:25200-29010, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:29120-30030, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:30750-36400, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:36670-37870, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:38250-38980, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:39100-40680, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:40920-41550, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:43230-45360, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:46100-46410, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:46460-49470, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:49710-50700, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:51050-52040, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:52580-56200, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:57250-57790, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:58320-59710, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:62156-66822, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:67047-68522, due to being absent in the computeMatrix output.\n",
      "Skipping MT:10376-10926, due to being absent in the computeMatrix output.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools/3.5.0;computeMatrix reference-point -S /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_Vector_LPS3.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_WT_LPS3.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_K197R_LPS3.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_K197Q_LPS3.bw  -R /data/akhtar/group2/rabbani/atac_project2078/hmmratac/ocrs/merged_peaks/merged_Vector_LPS3_.bed /data/akhtar/group2/rabbani/atac_project2078/hmmratac/ocrs/merged_peaks/merged_WT_LPS3_.bed /data/akhtar/group2/rabbani/atac_project2078/hmmratac/ocrs/merged_peaks/merged_K197R_LPS3_.bed /data/akhtar/group2/rabbani/atac_project2078/hmmratac/ocrs/merged_peaks/merged_K197Q_LPS3_.bed  -a 2000 -b 2000 --referencePoint center -p 20  --missingDataAsZero  --skipZeros  --samplesLabel Vector_LPS3 WT_LPS3 K197R_LPS3 K197Q_LPS3  --blackListFileName /data/repository/organisms/GRCm38_ensembl/akundaje/blacklist.bed  -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS3_onpeaks.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Skipping 1:58613950-58615190, due to being absent in the computeMatrix output.\n",
      "Skipping 1:88224040-88224530, due to being absent in the computeMatrix output.\n",
      "Skipping 1:183299140-183299590, due to being absent in the computeMatrix output.\n",
      "Skipping 11:109011640-109012100, due to being absent in the computeMatrix output.\n",
      "Skipping 13:119613020-119613530, due to being absent in the computeMatrix output.\n",
      "Skipping 13:119615910-119617480, due to being absent in the computeMatrix output.\n",
      "Skipping 15:75085700-75086100, due to being absent in the computeMatrix output.\n",
      "Skipping 15:75086260-75086530, due to being absent in the computeMatrix output.\n",
      "Skipping 15:75086670-75087040, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39842990-39843360, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39844861-39845251, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39846142-39846908, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39847720-39848835, due to being absent in the computeMatrix output.\n",
      "Skipping 18:68692070-68692290, due to being absent in the computeMatrix output.\n",
      "Skipping 2:98662240-98662900, due to being absent in the computeMatrix output.\n",
      "Skipping 5:146260990-146261360, due to being absent in the computeMatrix output.\n",
      "Skipping GL456210.1:125630-125920, due to being absent in the computeMatrix output.\n",
      "Skipping GL456210.1:141720-142670, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:36830-37120, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:56320-56790, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:139550-140370, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:174510-175120, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:125470-125690, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:129670-129990, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:150460-150660, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:151520-151740, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:153010-153380, due to being absent in the computeMatrix output.\n",
      "Skipping GL456216.1:13760-14030, due to being absent in the computeMatrix output.\n",
      "Skipping GL456216.1:15630-17540, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:36390-36680, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:64650-65500, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:85960-86270, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:162370-162700, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:38590-39510, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:138620-139500, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:159520-160060, due to being absent in the computeMatrix output.\n",
      "Skipping GL456239.1:33770-34600, due to being absent in the computeMatrix output.\n",
      "Skipping GL456239.1:37920-39790, due to being absent in the computeMatrix output.\n",
      "Skipping GL456359.1:19330-20080, due to being absent in the computeMatrix output.\n",
      "Skipping GL456372.1:12550-12770, due to being absent in the computeMatrix output.\n",
      "Skipping GL456372.1:17160-17610, due to being absent in the computeMatrix output.\n",
      "Skipping GL456372.1:20540-20820, due to being absent in the computeMatrix output.\n",
      "Skipping JH584295.1:1410-1850, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:260-530, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:680-1190, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:4010-5840, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:6070-7300, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:8930-10290, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:12690-12950, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:13030-13720, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:16120-16460, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:16900-17300, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:18790-19010, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:20620-21530, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:21920-22450, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:22550-22770, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:24000-24350, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:24440-24930, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:25270-25640, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:27810-28350, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:28510-28980, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:31070-33530, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:33790-36400, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:36670-36890, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:37010-37880, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:41140-41490, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:43110-45350, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:46660-47040, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:47120-49460, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:49790-50740, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:50900-52160, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:52760-55950, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:58130-59720, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:59820-61870, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:62187-66527, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:67057-68511, due to being absent in the computeMatrix output.\n",
      "Skipping MT:10326-10920, due to being absent in the computeMatrix output.\n",
      "Skipping 1:183299140-183299590, due to being absent in the computeMatrix output.\n",
      "Skipping 11:109011640-109012030, due to being absent in the computeMatrix output.\n",
      "Skipping 13:119613050-119613350, due to being absent in the computeMatrix output.\n",
      "Skipping 13:119615540-119617500, due to being absent in the computeMatrix output.\n",
      "Skipping 15:75086260-75086520, due to being absent in the computeMatrix output.\n",
      "Skipping 15:75086670-75087040, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39842990-39844140, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39844313-39846908, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39847720-39848840, due to being absent in the computeMatrix output.\n",
      "Skipping 18:68692070-68692300, due to being absent in the computeMatrix output.\n",
      "Skipping 2:98662240-98662480, due to being absent in the computeMatrix output.\n",
      "Skipping 2:98662510-98662900, due to being absent in the computeMatrix output.\n",
      "Skipping 5:146260990-146261350, due to being absent in the computeMatrix output.\n",
      "Skipping 6:103649080-103649280, due to being absent in the computeMatrix output.\n",
      "Skipping GL456210.1:122410-122750, due to being absent in the computeMatrix output.\n",
      "Skipping GL456210.1:125620-125940, due to being absent in the computeMatrix output.\n",
      "Skipping GL456210.1:137680-137920, due to being absent in the computeMatrix output.\n",
      "Skipping GL456210.1:141740-142820, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:36880-37340, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:56190-56750, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:139550-140380, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:174500-175000, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:123900-124630, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:125490-126330, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:129660-129990, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:150390-150680, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:151500-151740, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:152950-153410, due to being absent in the computeMatrix output.\n",
      "Skipping GL456216.1:13760-14030, due to being absent in the computeMatrix output.\n",
      "Skipping GL456216.1:15620-17480, due to being absent in the computeMatrix output.\n",
      "Skipping GL456216.1:32440-33030, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:36370-36880, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:48290-48530, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:64720-65500, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:85740-86270, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:38550-39510, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:138950-139310, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:159530-160160, due to being absent in the computeMatrix output.\n",
      "Skipping GL456239.1:37680-39780, due to being absent in the computeMatrix output.\n",
      "Skipping GL456359.1:19300-19870, due to being absent in the computeMatrix output.\n",
      "Skipping GL456366.1:28290-28680, due to being absent in the computeMatrix output.\n",
      "Skipping GL456372.1:12490-12810, due to being absent in the computeMatrix output.\n",
      "Skipping GL456372.1:17170-17700, due to being absent in the computeMatrix output.\n",
      "Skipping GL456372.1:20330-20830, due to being absent in the computeMatrix output.\n",
      "Skipping JH584295.1:1290-1960, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:280-510, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:680-1130, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:1500-3380, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:3980-5090, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:5450-6000, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:6130-7300, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:8930-10470, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:13420-13750, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:13810-14020, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:16910-17260, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:18790-19190, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:19280-19480, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:19840-20320, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:20810-21380, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:21880-22430, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:22520-22780, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:23990-25060, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:26060-29010, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:29380-29740, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:31050-36380, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:36520-40670, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:41100-41500, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:43110-45380, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:46460-47040, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:47070-49460, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:49730-50720, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:51650-52120, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:52750-55420, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:55470-56000, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:58130-59690, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:59835-61835, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:62145-66805, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:67058-68518, due to being absent in the computeMatrix output.\n",
      "Skipping MT:6504-7004, due to being absent in the computeMatrix output.\n",
      "Skipping MT:10340-10924, due to being absent in the computeMatrix output.\n",
      "Skipping 1:58613950-58615230, due to being absent in the computeMatrix output.\n",
      "Skipping 1:88223600-88224370, due to being absent in the computeMatrix output.\n",
      "Skipping 1:183299140-183299590, due to being absent in the computeMatrix output.\n",
      "Skipping 11:109011640-109012090, due to being absent in the computeMatrix output.\n",
      "Skipping 13:119613010-119613970, due to being absent in the computeMatrix output.\n",
      "Skipping 13:119616020-119617540, due to being absent in the computeMatrix output.\n",
      "Skipping 15:75085700-75086070, due to being absent in the computeMatrix output.\n",
      "Skipping 15:75086670-75087040, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39842990-39844150, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39844313-39846903, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39847720-39848840, due to being absent in the computeMatrix output.\n",
      "Skipping 18:68692070-68692280, due to being absent in the computeMatrix output.\n",
      "Skipping 2:98662240-98662470, due to being absent in the computeMatrix output.\n",
      "Skipping 2:98662510-98662870, due to being absent in the computeMatrix output.\n",
      "Skipping 5:146260990-146261350, due to being absent in the computeMatrix output.\n",
      "Skipping GL456210.1:122370-122640, due to being absent in the computeMatrix output.\n",
      "Skipping GL456210.1:125620-125950, due to being absent in the computeMatrix output.\n",
      "Skipping GL456210.1:141740-142770, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:36990-37330, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:56180-56750, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:116100-116520, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:139580-139840, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:141090-141320, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:174690-175000, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:123870-124940, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:125480-125970, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:152990-153380, due to being absent in the computeMatrix output.\n",
      "Skipping GL456216.1:13760-14030, due to being absent in the computeMatrix output.\n",
      "Skipping GL456216.1:15620-17480, due to being absent in the computeMatrix output.\n",
      "Skipping GL456216.1:34450-34810, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:36400-36880, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:48270-48530, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:64670-65510, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:85770-86270, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:38630-39510, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:138950-139410, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:159500-160130, due to being absent in the computeMatrix output.\n",
      "Skipping GL456239.1:33600-36300, due to being absent in the computeMatrix output.\n",
      "Skipping GL456239.1:37000-37540, due to being absent in the computeMatrix output.\n",
      "Skipping GL456239.1:37770-38170, due to being absent in the computeMatrix output.\n",
      "Skipping GL456239.1:38350-39810, due to being absent in the computeMatrix output.\n",
      "Skipping GL456359.1:19320-19800, due to being absent in the computeMatrix output.\n",
      "Skipping GL456366.1:28260-28570, due to being absent in the computeMatrix output.\n",
      "Skipping GL456372.1:12410-12930, due to being absent in the computeMatrix output.\n",
      "Skipping GL456372.1:17330-17550, due to being absent in the computeMatrix output.\n",
      "Skipping GL456372.1:20540-20820, due to being absent in the computeMatrix output.\n",
      "Skipping JH584295.1:1270-1730, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:310-510, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:680-1150, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:1500-3370, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:3990-5120, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:5450-7310, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:8850-10530, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:12240-12460, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:12680-13150, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:13430-13890, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:18770-19480, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:20140-21380, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:22560-22780, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:24410-25890, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:25950-28650, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:28750-29010, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:29310-29760, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:30740-33680, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:33840-35490, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:35690-36510, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:36650-40820, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:41140-41490, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:43150-45300, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:46660-49470, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:49540-51240, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:51390-52120, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:52600-55960, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:57460-57720, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:58230-59710, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:59830-61740, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:62176-64004, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:64076-66046, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:66164-66824, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:67056-68514, due to being absent in the computeMatrix output.\n",
      "Skipping MT:10400-10920, due to being absent in the computeMatrix output.\n",
      "Skipping 1:58613950-58615210, due to being absent in the computeMatrix output.\n",
      "Skipping 1:88223690-88224340, due to being absent in the computeMatrix output.\n",
      "Skipping 1:183299130-183299600, due to being absent in the computeMatrix output.\n",
      "Skipping 11:109011640-109012080, due to being absent in the computeMatrix output.\n",
      "Skipping 13:119613140-119613390, due to being absent in the computeMatrix output.\n",
      "Skipping 13:119616310-119616750, due to being absent in the computeMatrix output.\n",
      "Skipping 13:119617000-119617550, due to being absent in the computeMatrix output.\n",
      "Skipping 15:75086260-75086540, due to being absent in the computeMatrix output.\n",
      "Skipping 15:75086660-75087040, due to being absent in the computeMatrix output.\n",
      "Skipping 17:36231260-36232060, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39842990-39844140, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39844313-39844613, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39844660-39845750, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39846053-39846903, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39847720-39848840, due to being absent in the computeMatrix output.\n",
      "Skipping 18:68692070-68692270, due to being absent in the computeMatrix output.\n",
      "Skipping 2:98662230-98662480, due to being absent in the computeMatrix output.\n",
      "Skipping 2:98662510-98662890, due to being absent in the computeMatrix output.\n",
      "Skipping 5:146260990-146261360, due to being absent in the computeMatrix output.\n",
      "Skipping GL456210.1:125620-125940, due to being absent in the computeMatrix output.\n",
      "Skipping GL456210.1:141800-142780, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:36810-37330, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:56170-56750, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:139550-140140, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:174510-174990, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:123970-124920, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:125450-125970, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:150420-150700, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:151310-151570, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:153000-153430, due to being absent in the computeMatrix output.\n",
      "Skipping GL456216.1:13750-14030, due to being absent in the computeMatrix output.\n",
      "Skipping GL456216.1:15630-17540, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:36380-36700, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:48120-48560, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:64650-65510, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:85750-86270, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:162380-162580, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:38570-39510, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:127720-128540, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:138780-139450, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:159510-160070, due to being absent in the computeMatrix output.\n",
      "Skipping GL456239.1:22640-22960, due to being absent in the computeMatrix output.\n",
      "Skipping GL456239.1:33530-34550, due to being absent in the computeMatrix output.\n",
      "Skipping GL456239.1:37830-39750, due to being absent in the computeMatrix output.\n",
      "Skipping GL456359.1:19330-19630, due to being absent in the computeMatrix output.\n",
      "Skipping GL456359.1:19750-20070, due to being absent in the computeMatrix output.\n",
      "Skipping GL456366.1:28400-28630, due to being absent in the computeMatrix output.\n",
      "Skipping GL456372.1:12530-12780, due to being absent in the computeMatrix output.\n",
      "Skipping GL456372.1:17170-17810, due to being absent in the computeMatrix output.\n",
      "Skipping GL456372.1:20320-20830, due to being absent in the computeMatrix output.\n",
      "Skipping JH584295.1:1250-1960, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:280-520, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:670-3210, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:4010-5090, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:5510-5850, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:6230-7310, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:8850-10630, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:12260-12460, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:12680-12950, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:14070-14270, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:15090-15370, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:16140-16450, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:16890-17460, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:18790-18990, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:19100-19490, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:19820-20490, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:20680-21380, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:21920-22440, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:22520-22770, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:23980-24940, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:25050-25670, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:25940-28980, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:29330-30000, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:31020-33670, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:33810-36420, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:36670-37760, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:38180-40860, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:40960-41500, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:43220-45350, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:46620-50740, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:50800-51270, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:51380-52120, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:52580-55990, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:58120-59680, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:62162-66824, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:67054-68516, due to being absent in the computeMatrix output.\n",
      "Skipping MT:6543-6973, due to being absent in the computeMatrix output.\n",
      "Skipping MT:10336-10920, due to being absent in the computeMatrix output.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools/3.5.0;computeMatrix reference-point -S /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_Vector_LPS12.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_WT_LPS12.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_K197R_LPS12.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_K197Q_LPS12.bw  -R /data/akhtar/group2/rabbani/atac_project2078/hmmratac/ocrs/merged_peaks/merged_Vector_LPS12_.bed /data/akhtar/group2/rabbani/atac_project2078/hmmratac/ocrs/merged_peaks/merged_WT_LPS12_.bed /data/akhtar/group2/rabbani/atac_project2078/hmmratac/ocrs/merged_peaks/merged_K197R_LPS12_.bed /data/akhtar/group2/rabbani/atac_project2078/hmmratac/ocrs/merged_peaks/merged_K197Q_LPS12_.bed  -a 2000 -b 2000 --referencePoint center -p 20  --missingDataAsZero  --skipZeros  --samplesLabel Vector_LPS12 WT_LPS12 K197R_LPS12 K197Q_LPS12  --blackListFileName /data/repository/organisms/GRCm38_ensembl/akundaje/blacklist.bed  -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS12_onpeaks.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Skipping 1:88223390-88224370, due to being absent in the computeMatrix output.\n",
      "Skipping 1:183299130-183299590, due to being absent in the computeMatrix output.\n",
      "Skipping 11:109011640-109012030, due to being absent in the computeMatrix output.\n",
      "Skipping 13:119613030-119613850, due to being absent in the computeMatrix output.\n",
      "Skipping 13:119614180-119614780, due to being absent in the computeMatrix output.\n",
      "Skipping 13:119616320-119617610, due to being absent in the computeMatrix output.\n",
      "Skipping 15:75086660-75087040, due to being absent in the computeMatrix output.\n",
      "Skipping 17:36231260-36232060, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39842990-39844130, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39844313-39844623, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39844934-39845254, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39846613-39846903, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39847862-39848842, due to being absent in the computeMatrix output.\n",
      "Skipping 18:68692070-68692270, due to being absent in the computeMatrix output.\n",
      "Skipping 2:98662230-98662480, due to being absent in the computeMatrix output.\n",
      "Skipping 2:98662510-98662880, due to being absent in the computeMatrix output.\n",
      "Skipping 5:146260990-146261350, due to being absent in the computeMatrix output.\n",
      "Skipping GL456210.1:125630-125950, due to being absent in the computeMatrix output.\n",
      "Skipping GL456210.1:141760-142670, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:36770-37120, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:56200-56750, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:115860-116580, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:139550-140520, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:144190-144720, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:174510-175000, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:123880-124950, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:125470-126040, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:129660-130070, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:150460-150670, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:151440-151680, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:152940-153330, due to being absent in the computeMatrix output.\n",
      "Skipping GL456216.1:13760-14030, due to being absent in the computeMatrix output.\n",
      "Skipping GL456216.1:15620-17530, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:36380-36700, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:48140-48540, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:64430-65510, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:85610-86290, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:38590-39510, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:127650-128440, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:138680-139330, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:159470-160130, due to being absent in the computeMatrix output.\n",
      "Skipping GL456239.1:1560-1810, due to being absent in the computeMatrix output.\n",
      "Skipping GL456239.1:33550-36110, due to being absent in the computeMatrix output.\n",
      "Skipping GL456239.1:37530-39910, due to being absent in the computeMatrix output.\n",
      "Skipping GL456359.1:19320-19660, due to being absent in the computeMatrix output.\n",
      "Skipping GL456366.1:28300-28560, due to being absent in the computeMatrix output.\n",
      "Skipping GL456372.1:17210-17510, due to being absent in the computeMatrix output.\n",
      "Skipping GL456372.1:20320-20830, due to being absent in the computeMatrix output.\n",
      "Skipping GL456379.1:58320-58560, due to being absent in the computeMatrix output.\n",
      "Skipping GL456387.1:11170-11860, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:270-530, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:670-1020, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:1500-3160, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:3570-3820, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:4010-5260, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:5390-7350, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:8860-10530, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:12240-12460, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:12700-12950, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:13420-13720, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:15790-16010, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:16900-17280, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:19280-19480, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:19830-20160, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:20800-21270, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:22530-22780, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:23970-25050, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:25220-25610, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:26070-28990, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:29340-30020, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:31030-33680, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:33780-36430, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:36670-40680, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:41010-41550, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:43040-45380, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:46470-49460, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:49780-50700, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:50900-52000, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:52760-56030, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:57430-57750, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:58100-59560, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:60048-61768, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:62178-66926, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:67048-68518, due to being absent in the computeMatrix output.\n",
      "Skipping MT:6496-6976, due to being absent in the computeMatrix output.\n",
      "Skipping MT:10290-10926, due to being absent in the computeMatrix output.\n",
      "Skipping 1:24612796-24613056, due to being absent in the computeMatrix output.\n",
      "Skipping 1:88220390-88221090, due to being absent in the computeMatrix output.\n",
      "Skipping 1:183299140-183299600, due to being absent in the computeMatrix output.\n",
      "Skipping 11:109011640-109012030, due to being absent in the computeMatrix output.\n",
      "Skipping 13:119613020-119614000, due to being absent in the computeMatrix output.\n",
      "Skipping 13:119614150-119617610, due to being absent in the computeMatrix output.\n",
      "Skipping 15:75085690-75086080, due to being absent in the computeMatrix output.\n",
      "Skipping 15:75086260-75087040, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39842990-39843400, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39846685-39846905, due to being absent in the computeMatrix output.\n",
      "Skipping 18:68692070-68692270, due to being absent in the computeMatrix output.\n",
      "Skipping 2:98662240-98662480, due to being absent in the computeMatrix output.\n",
      "Skipping 5:146260990-146261350, due to being absent in the computeMatrix output.\n",
      "Skipping GL456210.1:125630-125930, due to being absent in the computeMatrix output.\n",
      "Skipping GL456210.1:141840-142830, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:36840-37120, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:56160-56760, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:116070-116580, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:139550-140360, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:174510-175000, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:123890-124910, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:125470-126080, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:129670-130040, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:153030-153310, due to being absent in the computeMatrix output.\n",
      "Skipping GL456216.1:13760-14030, due to being absent in the computeMatrix output.\n",
      "Skipping GL456216.1:15610-17470, due to being absent in the computeMatrix output.\n",
      "Skipping GL456216.1:34140-34340, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:36380-36680, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:48290-48530, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:64670-65500, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:85770-86270, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:38550-39510, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:138650-139320, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:159540-160070, due to being absent in the computeMatrix output.\n",
      "Skipping GL456239.1:1350-1720, due to being absent in the computeMatrix output.\n",
      "Skipping GL456239.1:33550-35230, due to being absent in the computeMatrix output.\n",
      "Skipping GL456239.1:35370-36440, due to being absent in the computeMatrix output.\n",
      "Skipping GL456239.1:36790-39930, due to being absent in the computeMatrix output.\n",
      "Skipping GL456359.1:19320-19840, due to being absent in the computeMatrix output.\n",
      "Skipping GL456366.1:28400-28700, due to being absent in the computeMatrix output.\n",
      "Skipping GL456372.1:17240-17600, due to being absent in the computeMatrix output.\n",
      "Skipping GL456372.1:20320-20820, due to being absent in the computeMatrix output.\n",
      "Skipping GL456387.1:11100-11600, due to being absent in the computeMatrix output.\n",
      "Skipping GL456387.1:11930-12220, due to being absent in the computeMatrix output.\n",
      "Skipping GL456387.1:14950-15410, due to being absent in the computeMatrix output.\n",
      "Skipping GL456387.1:16250-16530, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:270-510, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:680-3370, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:3570-3820, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:3980-5850, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:5940-7450, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:8850-10790, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:12250-12460, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:12680-13170, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:13380-13730, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:15790-16480, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:16900-17400, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:19080-19490, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:19820-20320, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:20430-21210, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:21740-22080, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:22280-22780, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:23970-25080, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:25260-29010, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:29300-30020, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:31020-38990, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:39120-40860, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:41050-41640, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:43230-45570, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:46460-50710, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:50950-52090, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:52590-56000, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:57250-57580, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:58200-59540, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:62178-66828, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:67058-68512, due to being absent in the computeMatrix output.\n",
      "Skipping MT:10341-10935, due to being absent in the computeMatrix output.\n",
      "Skipping 1:183299130-183299590, due to being absent in the computeMatrix output.\n",
      "Skipping 11:109011640-109012040, due to being absent in the computeMatrix output.\n",
      "Skipping 13:119613020-119613990, due to being absent in the computeMatrix output.\n",
      "Skipping 13:119614240-119614720, due to being absent in the computeMatrix output.\n",
      "Skipping 13:119615470-119617610, due to being absent in the computeMatrix output.\n",
      "Skipping 15:75086270-75086520, due to being absent in the computeMatrix output.\n",
      "Skipping 15:75086650-75087040, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39842990-39845890, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39846086-39846906, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39847720-39848840, due to being absent in the computeMatrix output.\n",
      "Skipping 18:68692070-68692270, due to being absent in the computeMatrix output.\n",
      "Skipping 2:98662240-98662480, due to being absent in the computeMatrix output.\n",
      "Skipping 2:98662510-98662880, due to being absent in the computeMatrix output.\n",
      "Skipping 5:146260990-146261350, due to being absent in the computeMatrix output.\n",
      "Skipping GL456210.1:125590-125940, due to being absent in the computeMatrix output.\n",
      "Skipping GL456210.1:141790-142850, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:56190-56750, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:139540-140390, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:174460-174950, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:123980-124900, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:125470-125960, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:129650-130050, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:153030-153300, due to being absent in the computeMatrix output.\n",
      "Skipping GL456216.1:13750-14020, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:36390-36670, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:64720-65510, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:85770-86270, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:38760-39510, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:138600-139370, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:159510-160060, due to being absent in the computeMatrix output.\n",
      "Skipping GL456239.1:33570-35890, due to being absent in the computeMatrix output.\n",
      "Skipping GL456239.1:37820-39860, due to being absent in the computeMatrix output.\n",
      "Skipping GL456359.1:19330-19560, due to being absent in the computeMatrix output.\n",
      "Skipping GL456372.1:20550-20820, due to being absent in the computeMatrix output.\n",
      "Skipping GL456379.1:58350-58680, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:280-510, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:680-3160, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:3990-5100, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:5930-7470, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:8840-10780, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:12260-12460, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:12670-12950, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:13380-13730, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:15780-16020, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:16130-16480, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:16880-17250, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:18770-18990, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:19280-19490, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:19820-20500, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:20670-21470, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:21740-22090, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:22350-22780, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:23990-25650, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:26080-27540, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:28490-29000, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:29310-29960, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:31040-36410, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:36690-37740, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:38150-40860, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:41010-41510, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:43070-45360, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:46680-49470, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:49560-50720, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:51060-51260, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:51390-52000, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:52760-54600, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:54760-56100, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:58090-59710, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:62175-66842, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:67056-68516, due to being absent in the computeMatrix output.\n",
      "Skipping MT:10280-10873, due to being absent in the computeMatrix output.\n",
      "Skipping 1:58613960-58615230, due to being absent in the computeMatrix output.\n",
      "Skipping 1:88220350-88221040, due to being absent in the computeMatrix output.\n",
      "Skipping 1:183299130-183299600, due to being absent in the computeMatrix output.\n",
      "Skipping 11:109011640-109012030, due to being absent in the computeMatrix output.\n",
      "Skipping 13:119609470-119609770, due to being absent in the computeMatrix output.\n",
      "Skipping 13:119613050-119613810, due to being absent in the computeMatrix output.\n",
      "Skipping 13:119614200-119614850, due to being absent in the computeMatrix output.\n",
      "Skipping 13:119615440-119617610, due to being absent in the computeMatrix output.\n",
      "Skipping 15:75086260-75086530, due to being absent in the computeMatrix output.\n",
      "Skipping 15:75086640-75087040, due to being absent in the computeMatrix output.\n",
      "Skipping 17:36231220-36232060, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39842990-39844060, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39846683-39846903, due to being absent in the computeMatrix output.\n",
      "Skipping 17:39848433-39848834, due to being absent in the computeMatrix output.\n",
      "Skipping 18:68692070-68692270, due to being absent in the computeMatrix output.\n",
      "Skipping 2:98662230-98662480, due to being absent in the computeMatrix output.\n",
      "Skipping 5:146260990-146261350, due to being absent in the computeMatrix output.\n",
      "Skipping GL456210.1:125640-125930, due to being absent in the computeMatrix output.\n",
      "Skipping GL456210.1:141730-142730, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:36820-37110, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:56310-56750, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:116140-116490, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:139550-140390, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:141090-141310, due to being absent in the computeMatrix output.\n",
      "Skipping GL456211.1:174530-174950, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:123880-124910, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:125510-125970, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:129660-130050, due to being absent in the computeMatrix output.\n",
      "Skipping GL456212.1:153010-153300, due to being absent in the computeMatrix output.\n",
      "Skipping GL456216.1:13750-14020, due to being absent in the computeMatrix output.\n",
      "Skipping GL456216.1:15630-17480, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:36380-36690, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:48150-48530, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:64730-65510, due to being absent in the computeMatrix output.\n",
      "Skipping GL456221.1:85610-86280, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:38560-39500, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:138610-139370, due to being absent in the computeMatrix output.\n",
      "Skipping GL456233.1:159530-160090, due to being absent in the computeMatrix output.\n",
      "Skipping GL456239.1:1380-1690, due to being absent in the computeMatrix output.\n",
      "Skipping GL456239.1:33520-39910, due to being absent in the computeMatrix output.\n",
      "Skipping GL456359.1:19320-19850, due to being absent in the computeMatrix output.\n",
      "Skipping GL456372.1:17250-17620, due to being absent in the computeMatrix output.\n",
      "Skipping GL456372.1:20550-20820, due to being absent in the computeMatrix output.\n",
      "Skipping GL456379.1:58350-58550, due to being absent in the computeMatrix output.\n",
      "Skipping GL456387.1:11130-11560, due to being absent in the computeMatrix output.\n",
      "Skipping GL456387.1:11690-12110, due to being absent in the computeMatrix output.\n",
      "Skipping GL456387.1:15040-15490, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:260-520, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:670-3240, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:3980-7350, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:7940-8150, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:8860-10480, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:12250-12460, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:12700-13160, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:13380-13780, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:13980-14260, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:15760-16540, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:16900-17320, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:18790-19490, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:19810-20330, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:20620-21230, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:21910-22790, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:23970-25880, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:26060-29020, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:29340-29770, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:30740-37830, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:37930-40850, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:40960-41590, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:42910-45370, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:46460-49470, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:49550-50710, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:50840-51310, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:51370-52120, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:52580-56080, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:58120-59550, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:60075-61695, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:62176-66835, due to being absent in the computeMatrix output.\n",
      "Skipping JH584304.1:67052-68512, due to being absent in the computeMatrix output.\n",
      "Skipping MT:6486-7006, due to being absent in the computeMatrix output.\n",
      "Skipping MT:10351-10931, due to being absent in the computeMatrix output.\n"
     ]
    }
   ],
   "source": [
    "path2peaks = os.path.join(\"/data/akhtar/group2/rabbani/atac_project2078/hmmratac/ocrs/merged_peaks\")\n",
    "for time in ['LPS0', 'LPS3', 'LPS12']:\n",
    "    bws = \"\"\n",
    "    names = \"\"\n",
    "    peaks = \"\"\n",
    "    for index, cond in enumerate([\"Vector\", \"WT\", \"K197R\", \"K197Q\"]):\n",
    "        bws += os.path.join(bw_path, \"_\".join([\"merged\", cond,time])+\".bw\")+\" \"\n",
    "        names += \"_\".join([cond,time])+ \" \"\n",
    "        peaks += os.path.join(path2peaks, \"_\".join([\"merged\", cond,time])+\"_.bed\")+\" \"\n",
    "    cmd = \"module load deeptools/3.5.0;\"\n",
    "    cmd += \"computeMatrix reference-point -S \"+bws\n",
    "    cmd += \" -R \"+peaks\n",
    "    cmd += \" -a 2000 -b 2000 --referencePoint center -p 20  --missingDataAsZero  --skipZeros \"\n",
    "    cmd += \" --samplesLabel \"+ names\n",
    "    cmd += \" --blackListFileName /data/repository/organisms/GRCm38_ensembl/akundaje/blacklist.bed \"\n",
    "    cmd += \" -o \"+os.path.join(output_path, time+\"_onpeaks.gz\")\n",
    "    print(cmd)\n",
    "    sp.check_output(cmd, shell = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "da34641d",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load ultraheatmap/1.3.1; addFeatureToMatrix -m /data/akhtar/group2/rabbani/atac_project2078/plots/LPS0_onpeaks.gz -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS0_onpeaks_mappedgenes.gz -t /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/K197Q_LPS0_vs_WT_LPS0.tsv /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/K197R_LPS0_vs_WT_LPS0.tsv /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/vector_LPS0_vs_WT_LPS0.tsv -g /data/manke/group/rabbani/rna_project2129/genes.filtered.gtf --annotationFeature 'gene' --featureIdColumn 'GeneID' --featureNames 'log2FoldChange' --referencePoint 'TSS' \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "***** WARNING: File /tmp/pybedtools.z4b_xbzp.tmp has inconsistent naming convention for record:\n",
      "CHR_MG153_PATCH\tHAVANA\tgene\t23936\t23936\t.\t-\t.\tgene_id \"ENSMUSG00000097830.3\"; gene_type \"protein_coding\"; gene_status \"KNOWN\"; gene_name \"Gstp1\"; level \"2\"; havana_gene \"OTTMUSG00000046062.1\";\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.z4b_xbzp.tmp has inconsistent naming convention for record:\n",
      "CHR_MG153_PATCH\tHAVANA\tgene\t23936\t23936\t.\t-\t.\tgene_id \"ENSMUSG00000097830.3\"; gene_type \"protein_coding\"; gene_status \"KNOWN\"; gene_name \"Gstp1\"; level \"2\"; havana_gene \"OTTMUSG00000046062.1\";\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load ultraheatmap/1.3.1; addFeatureToMatrix -m /data/akhtar/group2/rabbani/atac_project2078/plots/LPS3_onpeaks.gz -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS3_onpeaks_mappedgenes.gz -t /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/K197Q_LPS3_vs_WT_LPS3.tsv /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/K197R_LPS3_vs_WT_LPS3.tsv /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/vector_LPS3_vs_WT_LPS3.tsv -g /data/manke/group/rabbani/rna_project2129/genes.filtered.gtf --annotationFeature 'gene' --featureIdColumn 'GeneID' --featureNames 'log2FoldChange' --referencePoint 'TSS' \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "***** WARNING: File /tmp/pybedtools.x6n1ozmz.tmp has inconsistent naming convention for record:\n",
      "CHR_MG153_PATCH\tHAVANA\tgene\t23936\t23936\t.\t-\t.\tgene_id \"ENSMUSG00000097830.3\"; gene_type \"protein_coding\"; gene_status \"KNOWN\"; gene_name \"Gstp1\"; level \"2\"; havana_gene \"OTTMUSG00000046062.1\";\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.x6n1ozmz.tmp has inconsistent naming convention for record:\n",
      "CHR_MG153_PATCH\tHAVANA\tgene\t23936\t23936\t.\t-\t.\tgene_id \"ENSMUSG00000097830.3\"; gene_type \"protein_coding\"; gene_status \"KNOWN\"; gene_name \"Gstp1\"; level \"2\"; havana_gene \"OTTMUSG00000046062.1\";\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load ultraheatmap/1.3.1; addFeatureToMatrix -m /data/akhtar/group2/rabbani/atac_project2078/plots/LPS12_onpeaks.gz -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS12_onpeaks_mappedgenes.gz -t /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/K197Q_LPS12_vs_WT_LPS12.tsv /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/K197R_LPS12_vs_WT_LPS12.tsv /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/vector_LPS12_vs_WT_LPS12.tsv -g /data/manke/group/rabbani/rna_project2129/genes.filtered.gtf --annotationFeature 'gene' --featureIdColumn 'GeneID' --featureNames 'log2FoldChange' --referencePoint 'TSS' \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "***** WARNING: File /tmp/pybedtools.qr0gucoa.tmp has inconsistent naming convention for record:\n",
      "CHR_MG153_PATCH\tHAVANA\tgene\t23936\t23936\t.\t-\t.\tgene_id \"ENSMUSG00000097830.3\"; gene_type \"protein_coding\"; gene_status \"KNOWN\"; gene_name \"Gstp1\"; level \"2\"; havana_gene \"OTTMUSG00000046062.1\";\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.qr0gucoa.tmp has inconsistent naming convention for record:\n",
      "CHR_MG153_PATCH\tHAVANA\tgene\t23936\t23936\t.\t-\t.\tgene_id \"ENSMUSG00000097830.3\"; gene_type \"protein_coding\"; gene_status \"KNOWN\"; gene_name \"Gstp1\"; level \"2\"; havana_gene \"OTTMUSG00000046062.1\";\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for time in ['LPS0', 'LPS3', 'LPS12']:\n",
    "    cmd = \"module load ultraheatmap/1.3.1; addFeatureToMatrix \"\n",
    "    cmd += \"-m \"+os.path.join(output_path, time+\"_onpeaks.gz\")\n",
    "    cmd += \" -o \"+os.path.join(output_path, time+\"_onpeaks_mappedgenes.gz\")\n",
    "    cmd += \" -t \"+os.path.join(rna_path, \"K197Q_\"+time+\"_vs_WT_\"+time+\".tsv\")+ \" \"+os.path.join(rna_path, \"K197R_\"+time+\"_vs_WT_\"+time+\".tsv\")\n",
    "    cmd += \" \"+os.path.join(rna_path, \"vector_\"+time+\"_vs_WT_\"+time+\".tsv\")\n",
    "    cmd += \" -g \"+os.path.join(\"/data/manke/group/rabbani/rna_project2129/genes.filtered.gtf\")\n",
    "    cmd += \" --annotationFeature 'gene' --featureIdColumn 'GeneID' --featureNames 'log2FoldChange' --referencePoint 'TSS' \"\n",
    "    print(cmd)\n",
    "    sp.check_output(cmd, shell = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "f0204b11",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools/3.5.0; plotHeatmap  -m /data/akhtar/group2/rabbani/atac_project2078/plots/LPS0_onpeaks_mappedgenes.gz -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS0_onpeaks_mappedgenes.png --zMin 0 0 0 0 -2 -2 -2 --zMax 1 1 1 1 2 2 2  --colorMap Greens Greens Greens Greens RdBu_r RdBu_r RdBu_r --sortUsingSample 5 6 7 --samplesLabel vector_atac wt_atac kr_atac kq_atac kq_vs_wt kr_vs_wt vector_vs_wt \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper.py:1219: RuntimeWarning:\n",
      "\n",
      "Mean of empty slice\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"Greens\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"RdBu_r\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:738: MatplotlibDeprecationWarning:\n",
      "\n",
      "The 'alpha' parameter to Colorbar has no effect because it is overridden by the mappable; it is deprecated since 3.3 and will be removed two minor releases later.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:770: MatplotlibDeprecationWarning:\n",
      "\n",
      "savefig() got unexpected keyword argument \"pdd_inches\" which is no longer supported as of 3.3 and will become an error two minor releases later\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools/3.5.0; plotHeatmap  -m /data/akhtar/group2/rabbani/atac_project2078/plots/LPS3_onpeaks_mappedgenes.gz -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS3_onpeaks_mappedgenes.png --zMin 0 0 0 0 -2 -2 -2 --zMax 1 1 1 1 2 2 2  --colorMap Greens Greens Greens Greens RdBu_r RdBu_r RdBu_r --sortUsingSample 5 6 7 --samplesLabel vector_atac wt_atac kr_atac kq_atac kq_vs_wt kr_vs_wt vector_vs_wt \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper.py:1219: RuntimeWarning:\n",
      "\n",
      "Mean of empty slice\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"Greens\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"RdBu_r\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:738: MatplotlibDeprecationWarning:\n",
      "\n",
      "The 'alpha' parameter to Colorbar has no effect because it is overridden by the mappable; it is deprecated since 3.3 and will be removed two minor releases later.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:770: MatplotlibDeprecationWarning:\n",
      "\n",
      "savefig() got unexpected keyword argument \"pdd_inches\" which is no longer supported as of 3.3 and will become an error two minor releases later\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools/3.5.0; plotHeatmap  -m /data/akhtar/group2/rabbani/atac_project2078/plots/LPS12_onpeaks_mappedgenes.gz -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS12_onpeaks_mappedgenes.png --zMin 0 0 0 0 -2 -2 -2 --zMax 1 1 1 1 2 2 2  --colorMap Greens Greens Greens Greens RdBu_r RdBu_r RdBu_r --sortUsingSample 5 6 7 --samplesLabel vector_atac wt_atac kr_atac kq_atac kq_vs_wt kr_vs_wt vector_vs_wt \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper.py:1219: RuntimeWarning:\n",
      "\n",
      "Mean of empty slice\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"Greens\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"RdBu_r\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:738: MatplotlibDeprecationWarning:\n",
      "\n",
      "The 'alpha' parameter to Colorbar has no effect because it is overridden by the mappable; it is deprecated since 3.3 and will be removed two minor releases later.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:770: MatplotlibDeprecationWarning:\n",
      "\n",
      "savefig() got unexpected keyword argument \"pdd_inches\" which is no longer supported as of 3.3 and will become an error two minor releases later\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# plotheatmap\n",
    "for time in ['LPS0', 'LPS3', 'LPS12']:\n",
    "    cmd = \"module load deeptools/3.5.0; plotHeatmap \"\n",
    "    cmd += \" -m \"+os.path.join(output_path, time+\"_onpeaks_mappedgenes.gz\")\n",
    "    cmd += \" -o \"+os.path.join(output_path, time+\"_onpeaks_mappedgenes.png\")\n",
    "    cmd += \" --zMin 0 0 0 0 -2 -2 -2 --zMax 1 1 1 1 2 2 2 \" \n",
    "    cmd += \" --colorMap Greens Greens Greens Greens RdBu_r RdBu_r RdBu_r --sortUsingSample 5 6 7 --samplesLabel vector_atac wt_atac kr_atac kq_atac kq_vs_wt kr_vs_wt vector_vs_wt \"\n",
    "    print(cmd)\n",
    "    sp.check_output(cmd, shell = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "913f3927",
   "metadata": {},
   "outputs": [],
   "source": [
    "# # This is not fully done!\n",
    "# # similar approach as above ony on diff peaks\n",
    "# path2diffpeaks = os.path.join(\"/data/akhtar/group2/rabbani/atac_project2078/hmmratac/pairwise_comparison/wo_hiCov/deseq2\")\n",
    "# for time in ['LPS0', 'LPS3', 'LPS12']:\n",
    "#     for cond in [\"K197R\", \"K197Q\"]:\n",
    "#         df = pd.read_csv(os.path.join(path2diffpeaks, \"_\".join([cond,time, \"vs\", \"WT\", time])+\".tsv\"), sep = \"\\t\")\n",
    "#         up = df.loc[df[\"Status\"] == \"UP\"][[\"chr\", \"start\", \"end\"]]\n",
    "#         if len(up) != 0:\n",
    "#             up.to_csv(os.path.join(path2diffpeaks, \"diffPeaksBeds\", \"_\".join([cond,time])+\"_up.bed\"), sep = \"\\t\", header = None, index = False)\n",
    "#         down = df.loc[df[\"Status\"] == \"DOWN\"][[\"chr\", \"start\", \"end\"]]\n",
    "#         if len(down) != 0:\n",
    "#             down.to_csv(os.path.join(path2diffpeaks, \"diffPeaksBeds\", \"_\".join([cond,time])+\"_down.bed\"), sep = \"\\t\", header = None, index = False)\n",
    "# # This is not fully done!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "f9fab031",
   "metadata": {},
   "outputs": [],
   "source": [
    "blacklist= \"MT X Y JH584299.1 GL456233.1 JH584301.1 GL456211.1 GL456350.1 JH584293.1 GL456221.1 JH584297.1 JH584296.1 GL456354.1 JH584294.1 JH584298.1 JH584300.1 GL456219.1 GL456210.1 JH584303.1 JH584302.1 \"\\\n",
    "\"GL456212.1 JH584304.1 GL456379.1 GL456216.1 GL456393.1 GL456366.1 GL456367.1 GL456239.1 GL456213.1 GL456383.1 GL456385.1 GL456360.1 GL456378.1 GL456389.1 GL456372.1 GL456370.1 GL456381.1 GL456387.1 GL456390.1 \"\\\n",
    "\"GL456394.1 GL456392.1 GL456382.1 GL456359.1 GL456396.1 GL456368.1 JH584292.1 JH584295.1\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "b06e65d6",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS0_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS0_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS0_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS0_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS0_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS0_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS0_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS0_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS0_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS0_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS0_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS0_rep3.filtered.bw \n",
      "module load deeptools/3.5.0;multiBigwigSummary BED-file  -b /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS0_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS0_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS0_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS0_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS0_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS0_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS0_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS0_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS0_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS0_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS0_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS0_rep3.filtered.bw  -o /data/akhtar/group2/rabbani/atac_project2078/bwsummary/LPS0_vector_down.npz --BED /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/beds/vector_LPS0_vs_WT_LPS0_down.bed --chromosomesToSkip MT X Y JH584299.1 GL456233.1 JH584301.1 GL456211.1 GL456350.1 JH584293.1 GL456221.1 JH584297.1 JH584296.1 GL456354.1 JH584294.1 JH584298.1 JH584300.1 GL456219.1 GL456210.1 JH584303.1 JH584302.1 GL456212.1 JH584304.1 GL456379.1 GL456216.1 GL456393.1 GL456366.1 GL456367.1 GL456239.1 GL456213.1 GL456383.1 GL456385.1 GL456360.1 GL456378.1 GL456389.1 GL456372.1 GL456370.1 GL456381.1 GL456387.1 GL456390.1 GL456394.1 GL456392.1 GL456382.1 GL456359.1 GL456396.1 GL456368.1 JH584292.1 JH584295.1 -bl /data/repository/organisms/GRCm38_ensembl/akundaje/blacklist.bed  -p 15 --outRawCounts /data/akhtar/group2/rabbani/atac_project2078/bwsummary/LPS0_vector_down.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Number of bins found: 688\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools/3.5.0;multiBigwigSummary BED-file  -b /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS0_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS0_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS0_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS0_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS0_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS0_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS0_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS0_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS0_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS0_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS0_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS0_rep3.filtered.bw  -o /data/akhtar/group2/rabbani/atac_project2078/bwsummary/LPS0_vector_up.npz --BED /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/beds/vector_LPS0_vs_WT_LPS0_up.bed --chromosomesToSkip MT X Y JH584299.1 GL456233.1 JH584301.1 GL456211.1 GL456350.1 JH584293.1 GL456221.1 JH584297.1 JH584296.1 GL456354.1 JH584294.1 JH584298.1 JH584300.1 GL456219.1 GL456210.1 JH584303.1 JH584302.1 GL456212.1 JH584304.1 GL456379.1 GL456216.1 GL456393.1 GL456366.1 GL456367.1 GL456239.1 GL456213.1 GL456383.1 GL456385.1 GL456360.1 GL456378.1 GL456389.1 GL456372.1 GL456370.1 GL456381.1 GL456387.1 GL456390.1 GL456394.1 GL456392.1 GL456382.1 GL456359.1 GL456396.1 GL456368.1 JH584292.1 JH584295.1 -bl /data/repository/organisms/GRCm38_ensembl/akundaje/blacklist.bed  -p 15 --outRawCounts /data/akhtar/group2/rabbani/atac_project2078/bwsummary/LPS0_vector_up.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Number of bins found: 285\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools/3.5.0;multiBigwigSummary BED-file  -b /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS0_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS0_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS0_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS0_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS0_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS0_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS0_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS0_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS0_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS0_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS0_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS0_rep3.filtered.bw  -o /data/akhtar/group2/rabbani/atac_project2078/bwsummary/LPS0_K197R_down.npz --BED /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/beds/K197R_LPS0_vs_WT_LPS0_down.bed --chromosomesToSkip MT X Y JH584299.1 GL456233.1 JH584301.1 GL456211.1 GL456350.1 JH584293.1 GL456221.1 JH584297.1 JH584296.1 GL456354.1 JH584294.1 JH584298.1 JH584300.1 GL456219.1 GL456210.1 JH584303.1 JH584302.1 GL456212.1 JH584304.1 GL456379.1 GL456216.1 GL456393.1 GL456366.1 GL456367.1 GL456239.1 GL456213.1 GL456383.1 GL456385.1 GL456360.1 GL456378.1 GL456389.1 GL456372.1 GL456370.1 GL456381.1 GL456387.1 GL456390.1 GL456394.1 GL456392.1 GL456382.1 GL456359.1 GL456396.1 GL456368.1 JH584292.1 JH584295.1 -bl /data/repository/organisms/GRCm38_ensembl/akundaje/blacklist.bed  -p 15 --outRawCounts /data/akhtar/group2/rabbani/atac_project2078/bwsummary/LPS0_K197R_down.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Number of bins found: 514\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools/3.5.0;multiBigwigSummary BED-file  -b /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS0_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS0_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS0_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS0_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS0_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS0_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS0_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS0_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS0_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS0_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS0_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS0_rep3.filtered.bw  -o /data/akhtar/group2/rabbani/atac_project2078/bwsummary/LPS0_K197R_up.npz --BED /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/beds/K197R_LPS0_vs_WT_LPS0_up.bed --chromosomesToSkip MT X Y JH584299.1 GL456233.1 JH584301.1 GL456211.1 GL456350.1 JH584293.1 GL456221.1 JH584297.1 JH584296.1 GL456354.1 JH584294.1 JH584298.1 JH584300.1 GL456219.1 GL456210.1 JH584303.1 JH584302.1 GL456212.1 JH584304.1 GL456379.1 GL456216.1 GL456393.1 GL456366.1 GL456367.1 GL456239.1 GL456213.1 GL456383.1 GL456385.1 GL456360.1 GL456378.1 GL456389.1 GL456372.1 GL456370.1 GL456381.1 GL456387.1 GL456390.1 GL456394.1 GL456392.1 GL456382.1 GL456359.1 GL456396.1 GL456368.1 JH584292.1 JH584295.1 -bl /data/repository/organisms/GRCm38_ensembl/akundaje/blacklist.bed  -p 15 --outRawCounts /data/akhtar/group2/rabbani/atac_project2078/bwsummary/LPS0_K197R_up.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Number of bins found: 361\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools/3.5.0;multiBigwigSummary BED-file  -b /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS0_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS0_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS0_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS0_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS0_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS0_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS0_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS0_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS0_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS0_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS0_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS0_rep3.filtered.bw  -o /data/akhtar/group2/rabbani/atac_project2078/bwsummary/LPS0_K197Q_down.npz --BED /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/beds/K197Q_LPS0_vs_WT_LPS0_down.bed --chromosomesToSkip MT X Y JH584299.1 GL456233.1 JH584301.1 GL456211.1 GL456350.1 JH584293.1 GL456221.1 JH584297.1 JH584296.1 GL456354.1 JH584294.1 JH584298.1 JH584300.1 GL456219.1 GL456210.1 JH584303.1 JH584302.1 GL456212.1 JH584304.1 GL456379.1 GL456216.1 GL456393.1 GL456366.1 GL456367.1 GL456239.1 GL456213.1 GL456383.1 GL456385.1 GL456360.1 GL456378.1 GL456389.1 GL456372.1 GL456370.1 GL456381.1 GL456387.1 GL456390.1 GL456394.1 GL456392.1 GL456382.1 GL456359.1 GL456396.1 GL456368.1 JH584292.1 JH584295.1 -bl /data/repository/organisms/GRCm38_ensembl/akundaje/blacklist.bed  -p 15 --outRawCounts /data/akhtar/group2/rabbani/atac_project2078/bwsummary/LPS0_K197Q_down.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Number of bins found: 738\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools/3.5.0;multiBigwigSummary BED-file  -b /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS0_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS0_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS0_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS0_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS0_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS0_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS0_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS0_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS0_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS0_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS0_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS0_rep3.filtered.bw  -o /data/akhtar/group2/rabbani/atac_project2078/bwsummary/LPS0_K197Q_up.npz --BED /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/beds/K197Q_LPS0_vs_WT_LPS0_up.bed --chromosomesToSkip MT X Y JH584299.1 GL456233.1 JH584301.1 GL456211.1 GL456350.1 JH584293.1 GL456221.1 JH584297.1 JH584296.1 GL456354.1 JH584294.1 JH584298.1 JH584300.1 GL456219.1 GL456210.1 JH584303.1 JH584302.1 GL456212.1 JH584304.1 GL456379.1 GL456216.1 GL456393.1 GL456366.1 GL456367.1 GL456239.1 GL456213.1 GL456383.1 GL456385.1 GL456360.1 GL456378.1 GL456389.1 GL456372.1 GL456370.1 GL456381.1 GL456387.1 GL456390.1 GL456394.1 GL456392.1 GL456382.1 GL456359.1 GL456396.1 GL456368.1 JH584292.1 JH584295.1 -bl /data/repository/organisms/GRCm38_ensembl/akundaje/blacklist.bed  -p 15 --outRawCounts /data/akhtar/group2/rabbani/atac_project2078/bwsummary/LPS0_K197Q_up.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Number of bins found: 614\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS3_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS3_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS3_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS3_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS3_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS3_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS3_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS3_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS3_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS3_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS3_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS3_rep3.filtered.bw \n",
      "module load deeptools/3.5.0;multiBigwigSummary BED-file  -b /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS3_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS3_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS3_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS3_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS3_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS3_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS3_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS3_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS3_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS3_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS3_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS3_rep3.filtered.bw  -o /data/akhtar/group2/rabbani/atac_project2078/bwsummary/LPS3_vector_down.npz --BED /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/beds/vector_LPS3_vs_WT_LPS3_down.bed --chromosomesToSkip MT X Y JH584299.1 GL456233.1 JH584301.1 GL456211.1 GL456350.1 JH584293.1 GL456221.1 JH584297.1 JH584296.1 GL456354.1 JH584294.1 JH584298.1 JH584300.1 GL456219.1 GL456210.1 JH584303.1 JH584302.1 GL456212.1 JH584304.1 GL456379.1 GL456216.1 GL456393.1 GL456366.1 GL456367.1 GL456239.1 GL456213.1 GL456383.1 GL456385.1 GL456360.1 GL456378.1 GL456389.1 GL456372.1 GL456370.1 GL456381.1 GL456387.1 GL456390.1 GL456394.1 GL456392.1 GL456382.1 GL456359.1 GL456396.1 GL456368.1 JH584292.1 JH584295.1 -bl /data/repository/organisms/GRCm38_ensembl/akundaje/blacklist.bed  -p 15 --outRawCounts /data/akhtar/group2/rabbani/atac_project2078/bwsummary/LPS3_vector_down.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Number of bins found: 606\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools/3.5.0;multiBigwigSummary BED-file  -b /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS3_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS3_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS3_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS3_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS3_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS3_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS3_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS3_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS3_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS3_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS3_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS3_rep3.filtered.bw  -o /data/akhtar/group2/rabbani/atac_project2078/bwsummary/LPS3_vector_up.npz --BED /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/beds/vector_LPS3_vs_WT_LPS3_up.bed --chromosomesToSkip MT X Y JH584299.1 GL456233.1 JH584301.1 GL456211.1 GL456350.1 JH584293.1 GL456221.1 JH584297.1 JH584296.1 GL456354.1 JH584294.1 JH584298.1 JH584300.1 GL456219.1 GL456210.1 JH584303.1 JH584302.1 GL456212.1 JH584304.1 GL456379.1 GL456216.1 GL456393.1 GL456366.1 GL456367.1 GL456239.1 GL456213.1 GL456383.1 GL456385.1 GL456360.1 GL456378.1 GL456389.1 GL456372.1 GL456370.1 GL456381.1 GL456387.1 GL456390.1 GL456394.1 GL456392.1 GL456382.1 GL456359.1 GL456396.1 GL456368.1 JH584292.1 JH584295.1 -bl /data/repository/organisms/GRCm38_ensembl/akundaje/blacklist.bed  -p 15 --outRawCounts /data/akhtar/group2/rabbani/atac_project2078/bwsummary/LPS3_vector_up.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Number of bins found: 616\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools/3.5.0;multiBigwigSummary BED-file  -b /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS3_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS3_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS3_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS3_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS3_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS3_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS3_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS3_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS3_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS3_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS3_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS3_rep3.filtered.bw  -o /data/akhtar/group2/rabbani/atac_project2078/bwsummary/LPS3_K197R_down.npz --BED /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/beds/K197R_LPS3_vs_WT_LPS3_down.bed --chromosomesToSkip MT X Y JH584299.1 GL456233.1 JH584301.1 GL456211.1 GL456350.1 JH584293.1 GL456221.1 JH584297.1 JH584296.1 GL456354.1 JH584294.1 JH584298.1 JH584300.1 GL456219.1 GL456210.1 JH584303.1 JH584302.1 GL456212.1 JH584304.1 GL456379.1 GL456216.1 GL456393.1 GL456366.1 GL456367.1 GL456239.1 GL456213.1 GL456383.1 GL456385.1 GL456360.1 GL456378.1 GL456389.1 GL456372.1 GL456370.1 GL456381.1 GL456387.1 GL456390.1 GL456394.1 GL456392.1 GL456382.1 GL456359.1 GL456396.1 GL456368.1 JH584292.1 JH584295.1 -bl /data/repository/organisms/GRCm38_ensembl/akundaje/blacklist.bed  -p 15 --outRawCounts /data/akhtar/group2/rabbani/atac_project2078/bwsummary/LPS3_K197R_down.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Number of bins found: 184\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools/3.5.0;multiBigwigSummary BED-file  -b /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS3_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS3_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS3_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS3_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS3_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS3_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS3_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS3_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS3_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS3_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS3_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS3_rep3.filtered.bw  -o /data/akhtar/group2/rabbani/atac_project2078/bwsummary/LPS3_K197R_up.npz --BED /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/beds/K197R_LPS3_vs_WT_LPS3_up.bed --chromosomesToSkip MT X Y JH584299.1 GL456233.1 JH584301.1 GL456211.1 GL456350.1 JH584293.1 GL456221.1 JH584297.1 JH584296.1 GL456354.1 JH584294.1 JH584298.1 JH584300.1 GL456219.1 GL456210.1 JH584303.1 JH584302.1 GL456212.1 JH584304.1 GL456379.1 GL456216.1 GL456393.1 GL456366.1 GL456367.1 GL456239.1 GL456213.1 GL456383.1 GL456385.1 GL456360.1 GL456378.1 GL456389.1 GL456372.1 GL456370.1 GL456381.1 GL456387.1 GL456390.1 GL456394.1 GL456392.1 GL456382.1 GL456359.1 GL456396.1 GL456368.1 JH584292.1 JH584295.1 -bl /data/repository/organisms/GRCm38_ensembl/akundaje/blacklist.bed  -p 15 --outRawCounts /data/akhtar/group2/rabbani/atac_project2078/bwsummary/LPS3_K197R_up.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Number of bins found: 197\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools/3.5.0;multiBigwigSummary BED-file  -b /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS3_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS3_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS3_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS3_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS3_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS3_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS3_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS3_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS3_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS3_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS3_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS3_rep3.filtered.bw  -o /data/akhtar/group2/rabbani/atac_project2078/bwsummary/LPS3_K197Q_down.npz --BED /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/beds/K197Q_LPS3_vs_WT_LPS3_down.bed --chromosomesToSkip MT X Y JH584299.1 GL456233.1 JH584301.1 GL456211.1 GL456350.1 JH584293.1 GL456221.1 JH584297.1 JH584296.1 GL456354.1 JH584294.1 JH584298.1 JH584300.1 GL456219.1 GL456210.1 JH584303.1 JH584302.1 GL456212.1 JH584304.1 GL456379.1 GL456216.1 GL456393.1 GL456366.1 GL456367.1 GL456239.1 GL456213.1 GL456383.1 GL456385.1 GL456360.1 GL456378.1 GL456389.1 GL456372.1 GL456370.1 GL456381.1 GL456387.1 GL456390.1 GL456394.1 GL456392.1 GL456382.1 GL456359.1 GL456396.1 GL456368.1 JH584292.1 JH584295.1 -bl /data/repository/organisms/GRCm38_ensembl/akundaje/blacklist.bed  -p 15 --outRawCounts /data/akhtar/group2/rabbani/atac_project2078/bwsummary/LPS3_K197Q_down.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Number of bins found: 575\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools/3.5.0;multiBigwigSummary BED-file  -b /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS3_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS3_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS3_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS3_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS3_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS3_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS3_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS3_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS3_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS3_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS3_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS3_rep3.filtered.bw  -o /data/akhtar/group2/rabbani/atac_project2078/bwsummary/LPS3_K197Q_up.npz --BED /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/beds/K197Q_LPS3_vs_WT_LPS3_up.bed --chromosomesToSkip MT X Y JH584299.1 GL456233.1 JH584301.1 GL456211.1 GL456350.1 JH584293.1 GL456221.1 JH584297.1 JH584296.1 GL456354.1 JH584294.1 JH584298.1 JH584300.1 GL456219.1 GL456210.1 JH584303.1 JH584302.1 GL456212.1 JH584304.1 GL456379.1 GL456216.1 GL456393.1 GL456366.1 GL456367.1 GL456239.1 GL456213.1 GL456383.1 GL456385.1 GL456360.1 GL456378.1 GL456389.1 GL456372.1 GL456370.1 GL456381.1 GL456387.1 GL456390.1 GL456394.1 GL456392.1 GL456382.1 GL456359.1 GL456396.1 GL456368.1 JH584292.1 JH584295.1 -bl /data/repository/organisms/GRCm38_ensembl/akundaje/blacklist.bed  -p 15 --outRawCounts /data/akhtar/group2/rabbani/atac_project2078/bwsummary/LPS3_K197Q_up.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Number of bins found: 355\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS12_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS12_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS12_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS12_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS12_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS12_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS12_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS12_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS12_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS12_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS12_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS12_rep3.filtered.bw \n",
      "module load deeptools/3.5.0;multiBigwigSummary BED-file  -b /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS12_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS12_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS12_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS12_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS12_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS12_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS12_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS12_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS12_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS12_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS12_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS12_rep3.filtered.bw  -o /data/akhtar/group2/rabbani/atac_project2078/bwsummary/LPS12_vector_down.npz --BED /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/beds/vector_LPS12_vs_WT_LPS12_down.bed --chromosomesToSkip MT X Y JH584299.1 GL456233.1 JH584301.1 GL456211.1 GL456350.1 JH584293.1 GL456221.1 JH584297.1 JH584296.1 GL456354.1 JH584294.1 JH584298.1 JH584300.1 GL456219.1 GL456210.1 JH584303.1 JH584302.1 GL456212.1 JH584304.1 GL456379.1 GL456216.1 GL456393.1 GL456366.1 GL456367.1 GL456239.1 GL456213.1 GL456383.1 GL456385.1 GL456360.1 GL456378.1 GL456389.1 GL456372.1 GL456370.1 GL456381.1 GL456387.1 GL456390.1 GL456394.1 GL456392.1 GL456382.1 GL456359.1 GL456396.1 GL456368.1 JH584292.1 JH584295.1 -bl /data/repository/organisms/GRCm38_ensembl/akundaje/blacklist.bed  -p 15 --outRawCounts /data/akhtar/group2/rabbani/atac_project2078/bwsummary/LPS12_vector_down.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Number of bins found: 459\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools/3.5.0;multiBigwigSummary BED-file  -b /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS12_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS12_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS12_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS12_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS12_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS12_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS12_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS12_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS12_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS12_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS12_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS12_rep3.filtered.bw  -o /data/akhtar/group2/rabbani/atac_project2078/bwsummary/LPS12_vector_up.npz --BED /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/beds/vector_LPS12_vs_WT_LPS12_up.bed --chromosomesToSkip MT X Y JH584299.1 GL456233.1 JH584301.1 GL456211.1 GL456350.1 JH584293.1 GL456221.1 JH584297.1 JH584296.1 GL456354.1 JH584294.1 JH584298.1 JH584300.1 GL456219.1 GL456210.1 JH584303.1 JH584302.1 GL456212.1 JH584304.1 GL456379.1 GL456216.1 GL456393.1 GL456366.1 GL456367.1 GL456239.1 GL456213.1 GL456383.1 GL456385.1 GL456360.1 GL456378.1 GL456389.1 GL456372.1 GL456370.1 GL456381.1 GL456387.1 GL456390.1 GL456394.1 GL456392.1 GL456382.1 GL456359.1 GL456396.1 GL456368.1 JH584292.1 JH584295.1 -bl /data/repository/organisms/GRCm38_ensembl/akundaje/blacklist.bed  -p 15 --outRawCounts /data/akhtar/group2/rabbani/atac_project2078/bwsummary/LPS12_vector_up.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Number of bins found: 493\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools/3.5.0;multiBigwigSummary BED-file  -b /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS12_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS12_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS12_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS12_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS12_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS12_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS12_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS12_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS12_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS12_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS12_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS12_rep3.filtered.bw  -o /data/akhtar/group2/rabbani/atac_project2078/bwsummary/LPS12_K197Q_down.npz --BED /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/beds/K197Q_LPS12_vs_WT_LPS12_down.bed --chromosomesToSkip MT X Y JH584299.1 GL456233.1 JH584301.1 GL456211.1 GL456350.1 JH584293.1 GL456221.1 JH584297.1 JH584296.1 GL456354.1 JH584294.1 JH584298.1 JH584300.1 GL456219.1 GL456210.1 JH584303.1 JH584302.1 GL456212.1 JH584304.1 GL456379.1 GL456216.1 GL456393.1 GL456366.1 GL456367.1 GL456239.1 GL456213.1 GL456383.1 GL456385.1 GL456360.1 GL456378.1 GL456389.1 GL456372.1 GL456370.1 GL456381.1 GL456387.1 GL456390.1 GL456394.1 GL456392.1 GL456382.1 GL456359.1 GL456396.1 GL456368.1 JH584292.1 JH584295.1 -bl /data/repository/organisms/GRCm38_ensembl/akundaje/blacklist.bed  -p 15 --outRawCounts /data/akhtar/group2/rabbani/atac_project2078/bwsummary/LPS12_K197Q_down.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Number of bins found: 786\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools/3.5.0;multiBigwigSummary BED-file  -b /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS12_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS12_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/Vector_LPS12_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS12_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS12_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/WT_LPS12_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS12_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS12_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197R_LPS12_rep3.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS12_rep1.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS12_rep2.filtered.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline/BigWigs/K197Q_LPS12_rep3.filtered.bw  -o /data/akhtar/group2/rabbani/atac_project2078/bwsummary/LPS12_K197Q_up.npz --BED /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/beds/K197Q_LPS12_vs_WT_LPS12_up.bed --chromosomesToSkip MT X Y JH584299.1 GL456233.1 JH584301.1 GL456211.1 GL456350.1 JH584293.1 GL456221.1 JH584297.1 JH584296.1 GL456354.1 JH584294.1 JH584298.1 JH584300.1 GL456219.1 GL456210.1 JH584303.1 JH584302.1 GL456212.1 JH584304.1 GL456379.1 GL456216.1 GL456393.1 GL456366.1 GL456367.1 GL456239.1 GL456213.1 GL456383.1 GL456385.1 GL456360.1 GL456378.1 GL456389.1 GL456372.1 GL456370.1 GL456381.1 GL456387.1 GL456390.1 GL456394.1 GL456392.1 GL456382.1 GL456359.1 GL456396.1 GL456368.1 JH584292.1 JH584295.1 -bl /data/repository/organisms/GRCm38_ensembl/akundaje/blacklist.bed  -p 15 --outRawCounts /data/akhtar/group2/rabbani/atac_project2078/bwsummary/LPS12_K197Q_up.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Number of bins found: 1702\n"
     ]
    }
   ],
   "source": [
    "# bigwigsummary of atac signal on diff genes\n",
    "for time in ['LPS0', 'LPS3', 'LPS12']:\n",
    "    bws = \"\"\n",
    "    \n",
    "    for cond in [\"Vector\", \"WT\", \"K197R\", \"K197Q\"]:\n",
    "        for rep in ['rep1', 'rep2', 'rep3']:\n",
    "            bws += os.path.join(root, \"atac_pipeline\", \"BigWigs\", \"_\".join([cond, time, rep])+\".filtered.bw\")+\" \"\n",
    "    print(bws)\n",
    "    for cond in [\"vector\", \"K197R\", \"K197Q\"]:\n",
    "        beds = {\"down\":os.path.join(rna_path, \"beds\", \"_\".join([cond,time,\"vs\", \"WT\",time])+\"_down.bed\"),\n",
    "                \"up\":os.path.join(rna_path, \"beds\", \"_\".join([cond,time,\"vs\", \"WT\",time])+\"_up.bed\")}\n",
    "        for name, bed in beds.items():\n",
    "            if os.path.isfile(bed):\n",
    "                cmd = \"module load deeptools/3.5.0;\"\n",
    "                cmd += \"multiBigwigSummary BED-file \"   \n",
    "                cmd += \" -b \"+bws\n",
    "                cmd += \" -o \"+os.path.join(\"/data/akhtar/group2/rabbani/atac_project2078/bwsummary\", \"_\".join([time,cond,name])+\".npz\")\n",
    "                cmd +=  \" --BED \"+ bed\n",
    "                cmd += \" --chromosomesToSkip \"+blacklist\n",
    "                cmd += \" -bl /data/repository/organisms/GRCm38_ensembl/akundaje/blacklist.bed \"\n",
    "                cmd += \" -p 15 --outRawCounts \"+os.path.join(\"/data/akhtar/group2/rabbani/atac_project2078/bwsummary\", \"_\".join([time,cond,name])+\".tsv\")\n",
    "                print(cmd)\n",
    "                sp.check_output(cmd, shell = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "01eb3925",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ultraheatmap on all active TSS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "75328143",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Make a list of active genes, those which are available in deseq result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "ca0bec26",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/localenv/rabbani/anaconda/miniconda3/envs/matplotlib/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3444: DtypeWarning: Columns (0) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  exec(code_obj, self.user_global_ns, self.user_ns)\n"
     ]
    }
   ],
   "source": [
    "# make the region bed file\n",
    "gene_coord = os.path.join(\"/data/manke/group/rabbani/rna_project2129/genes.filtered.gtf\")\n",
    "gene_coord = pd.read_csv(gene_coord, sep = \"\\t\", header = None, comment = \"#\")\n",
    "gene_coord = gene_coord.loc[gene_coord[2] == \"gene\"]\n",
    "gene_coord[8] = gene_coord[8].str.split('\"', 2, expand=True)[1]\n",
    "gene_coord = gene_coord[[0,3,4,8,5,6]]\n",
    "rna_file = pd.read_csv(os.path.join(rna_path, \"K197R_LPS0_vs_WT_LPS0.tsv\"), sep = \"\\t\") # any tsv file of this path can be used, they all contain the same list of genes\n",
    "regions_to_plot = rna_file.merge(gene_coord, left_on = \"GeneID\", right_on = gene_coord[8], how = \"inner\")\n",
    "regions_to_plot[[0,3,4,8,5,6]].to_csv(os.path.join(rna_path, \"activegenes.bed\"), sep = \"\\t\", header = None, index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "3bfd1579",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools/3.5.0;computeMatrix reference-point -S /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_Vector_LPS0.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_WT_LPS0.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_K197R_LPS0.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_K197Q_LPS0.bw  -R /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/activegenes.bed -a 2000 -b 2000 --referencePoint TSS -p 20  --missingDataAsZero  --skipZeros  --samplesLabel Vector_LPS0 WT_LPS0 K197R_LPS0 K197Q_LPS0  --blackListFileName /data/repository/organisms/GRCm38_ensembl/akundaje/blacklist.bed  -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS0_onMouseTSS.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Skipping ENSMUSG00000000134.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000000223.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000000355.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000000787.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000000838.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000000881.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000001127.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000001128.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000001131.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000001134.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000001173.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000001924.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000001962.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000001964.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000002010.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000002014.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000002015.5, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000004221.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000004317.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000006373.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000006423.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000006678.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000008035.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000008682.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000009406.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000009731.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000009995.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000015214.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000015217.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000015289.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000015290.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000015291.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000015340.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000015342.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000015668.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000016239.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000016308.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000016319.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000016382.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000016409.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000016427.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000016534.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000017057.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000019087.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000019088.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000019558.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000020457.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000023074.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000023764.18, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025037.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025040.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025058.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025059.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025246.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025257.5, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025260.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025261.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025262.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025264.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025266.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025268.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025269.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025271.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025283.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025287.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025289.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025332.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025525.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025531.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025626.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025630.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025742.5, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025860.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025862.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031007.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031010.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031012.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031059.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031060.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031065.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031066.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031079.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031093.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031095.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031101.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031103.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031104.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031105.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031107.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031109.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031125.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031133.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031134.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031137.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031138.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031142.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031143.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031144.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031145.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031146.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031147.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031148.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031149.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031150.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031153.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031154.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031155.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031156.18, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031157.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031158.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031161.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031165.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031166.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031167.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031168.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031169.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031171.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031174.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031176.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031196.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031197.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031198.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031200.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031201.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031207.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031214.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031216.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031217.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031220.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031221.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031226.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031227.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031229.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031231.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031232.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031242.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031245.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031246.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031256.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031257.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031258.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031262.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031264.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031266.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031273.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031274.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031278.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031289.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031292.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031295.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031299.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031303.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031304.18, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031309.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031310.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031311.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031312.5, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031314.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031320.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031328.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031333.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031337.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031347.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031349.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031352.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031353.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031357.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031358.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031360.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031365.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031367.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031370.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031371.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031373.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031378.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031381.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031382.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031383.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031385.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031386.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031387.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031388.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031389.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031391.18, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031392.18, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031393.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031398.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031399.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031400.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031402.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031403.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031410.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031422.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031425.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031429.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031431.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031432.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031433.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031434.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031438.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000032750.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000032806.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000033777.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000033792.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000033983.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000034055.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000034160.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000034248.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000034311.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000034403.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000034457.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000034480.18, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000035062.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000035150.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000035232.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000035299.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000035476.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000035725.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000035776.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000035847.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000035967.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000036022.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000036131.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000036537.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000036572.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000036769.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000036916.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000036932.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000036959.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000036985.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000037217.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000037315.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000037341.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000037347.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000037369.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000037475.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000037636.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000038344.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000039201.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000039231.18, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000039382.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000039545.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000039556.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000040363.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000040522.5, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000040586.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000040621.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000040749.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000040855.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000040990.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000041096.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000041115.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000041133.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000041229.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000041353.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000041552.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000041633.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000041649.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000041658.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000041718.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000042225.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000042271.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000042473.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000042498.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000042595.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000042712.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000042903.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000043384.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000043929.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000044148.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000044149.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000044150.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000044348.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000044583.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000045237.5, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000045427.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000045694.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000046032.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000046269.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000046432.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000046873.18, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000046993.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000047045.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000047344.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000047694.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000047757.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000048007.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000048047.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000048592.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000048600.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000048621.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000048949.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000048970.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000049775.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000050029.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000050148.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000050232.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000050332.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000050379.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000050900.5, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000051220.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000051224.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000051412.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000051537.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000052364.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000052676.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000052821.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000054293.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000054626.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000054737.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000055188.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000056537.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000056673.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000057421.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000057457.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000059326.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000059334.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000059461.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000059708.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000060090.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000060681.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000061273.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000061778.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000062070.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000062783.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000062949.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000063018.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000063663.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000063785.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000063816.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000063897.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064127.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064337.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064339.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064340.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064341.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064343.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064344.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064345.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064346.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064351.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064354.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064356.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064357.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064358.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064360.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064363.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064367.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064368.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064370.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064371.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064372.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000065422.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000065642.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000065947.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000067194.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000067369.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000067567.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000067608.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000067873.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000067924.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000067925.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000068243.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000068457.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000069045.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000069049.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000071708.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000071748.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000072844.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000072969.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000073008.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000073010.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000073016.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000073060.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000073062.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000073130.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000073131.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000073139.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000073176.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000073274.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000074527.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000075359.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000076258.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000078122.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000078314.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000078317.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000078606.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000078894.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000078897.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079190.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079192.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079222.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079297.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079316.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079317.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079418.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079435.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079480.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079481.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079487.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079508.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079509.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079641.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079794.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079808.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079834.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000080718.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000080774.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000080775.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000080845.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000080877.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000080896.5, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000080923.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000080942.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000080968.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000080974.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081004.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081049.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081137.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081145.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081159.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081187.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081221.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081232.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081235.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081281.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081289.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081350.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081373.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081418.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081603.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081620.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081684.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081718.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081729.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081739.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081752.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082070.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082076.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082109.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082121.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082154.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082189.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082197.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082245.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082260.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082284.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082317.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082319.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082336.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082383.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082423.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082625.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082655.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082675.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082678.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082693.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082718.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082729.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082755.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082776.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082778.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082838.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082905.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082926.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082927.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083011.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083019.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083033.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083089.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083199.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083200.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083258.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083283.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083307.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083311.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083327.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083362.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083364.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083367.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083411.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083621.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083646.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083668.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083670.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083736.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083754.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083798.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083823.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083841.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083892.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083906.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083960.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084131.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084166.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084216.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084284.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084289.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084323.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084331.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084348.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084349.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084361.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084616.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084771.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084830.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084932.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000085314.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000085396.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000085587.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000085611.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000085711.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000086049.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000086316.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000086370.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000086404.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000086629.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000086691.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000086695.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000086736.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000086877.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000087159.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000087174.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000087235.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000087252.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000087365.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000087368.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000087396.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000087403.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000087424.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000087569.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000090110.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000090141.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000094054.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000094514.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000094621.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000094728.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000094799.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000095041.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000095092.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000095180.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000095500.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000095595.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000095597.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000095672.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000095742.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000096056.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000096100.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000096768.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000097468.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000097497.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000097571.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000097971.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000098178.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000098559.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000098973.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000099876.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000100104.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000101499.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000101939.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000102426.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000103436.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000103697.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000106106.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000109459.1, due to being absent in the computeMatrix output.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools/3.5.0;computeMatrix reference-point -S /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_Vector_LPS3.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_WT_LPS3.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_K197R_LPS3.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_K197Q_LPS3.bw  -R /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/activegenes.bed -a 2000 -b 2000 --referencePoint TSS -p 20  --missingDataAsZero  --skipZeros  --samplesLabel Vector_LPS3 WT_LPS3 K197R_LPS3 K197Q_LPS3  --blackListFileName /data/repository/organisms/GRCm38_ensembl/akundaje/blacklist.bed  -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS3_onMouseTSS.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Skipping ENSMUSG00000000134.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000000223.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000000355.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000000787.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000000838.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000000881.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000001127.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000001128.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000001131.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000001134.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000001173.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000001924.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000001962.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000001964.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000002010.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000002014.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000002015.5, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000004221.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000004317.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000006373.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000006423.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000006678.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000008035.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000008682.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000009406.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000009731.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000009995.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000015214.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000015217.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000015289.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000015290.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000015291.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000015340.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000015342.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000015668.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000016239.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000016308.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000016319.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000016382.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000016409.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000016427.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000016534.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000017057.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000019087.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000019088.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000019558.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000020457.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000023074.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000023764.18, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025037.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025040.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025058.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025059.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025246.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025257.5, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025260.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025261.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025262.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025264.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025266.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025268.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025269.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025271.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025283.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025287.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025289.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025332.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025525.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025531.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025626.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025630.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025742.5, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025860.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025862.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031007.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031010.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031012.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031059.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031060.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031065.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031066.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031079.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031093.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031095.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031101.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031103.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031104.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031105.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031107.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031109.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031125.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031133.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031134.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031137.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031138.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031142.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031143.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031144.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031145.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031146.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031147.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031148.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031149.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031150.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031153.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031154.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031155.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031156.18, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031157.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031158.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031161.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031165.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031166.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031167.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031168.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031169.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031171.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031174.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031176.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031196.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031197.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031198.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031200.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031201.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031207.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031214.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031216.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031217.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031220.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031221.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031226.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031227.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031229.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031231.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031232.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031242.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031245.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031246.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031256.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031257.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031258.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031262.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031264.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031266.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031273.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031274.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031278.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031289.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031292.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031295.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031299.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031303.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031304.18, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031309.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031310.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031311.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031312.5, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031314.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031320.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031328.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031333.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031337.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031347.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031349.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031352.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031353.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031357.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031358.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031360.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031365.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031367.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031370.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031371.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031373.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031378.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031381.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031382.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031383.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031385.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031386.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031387.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031388.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031389.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031391.18, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031392.18, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031393.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031398.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031399.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031400.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031402.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031403.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031410.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031422.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031425.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031429.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031431.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031432.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031433.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031434.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031438.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000032750.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000032806.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000033777.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000033792.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000033983.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000034055.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000034160.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000034248.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000034311.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000034403.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000034457.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000034480.18, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000035062.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000035150.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000035232.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000035299.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000035476.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000035725.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000035776.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000035847.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000035967.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000036022.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000036131.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000036537.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000036572.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000036769.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000036916.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000036932.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000036959.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000036985.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000037217.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000037315.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000037341.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000037347.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000037369.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000037475.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000037636.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000038344.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000039201.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000039231.18, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000039382.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000039545.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000039556.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000040363.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000040522.5, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000040586.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000040621.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000040749.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000040855.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000040990.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000041096.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000041115.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000041133.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000041229.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000041353.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000041552.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000041633.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000041649.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000041658.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000041718.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000042225.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000042271.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000042473.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000042498.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000042595.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000042712.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000042903.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000043384.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000043929.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000044148.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000044149.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000044150.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000044348.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000044583.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000045237.5, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000045427.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000045694.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000046032.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000046269.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000046432.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000046873.18, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000046993.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000047045.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000047344.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000047694.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000047757.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000048007.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000048047.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000048592.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000048600.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000048621.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000048949.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000048970.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000049775.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000050029.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000050148.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000050232.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000050332.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000050379.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000050900.5, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000051220.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000051224.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000051412.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000051537.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000052364.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000052676.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000052821.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000053038.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000054293.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000054626.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000054737.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000055188.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000056537.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000056673.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000057421.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000057457.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000059326.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000059334.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000059461.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000059708.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000060090.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000060681.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000061273.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000061778.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000062070.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000062783.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000062949.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000063018.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000063663.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000063785.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000063816.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000063897.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064127.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064337.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064339.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064340.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064341.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064343.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064344.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064345.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064346.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064351.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064354.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064356.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064357.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064358.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064360.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064363.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064367.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064368.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064370.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064371.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064372.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000065422.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000065642.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000065947.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000067194.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000067369.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000067567.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000067873.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000067924.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000067925.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000068243.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000068457.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000069045.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000069049.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000071708.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000071748.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000072844.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000072969.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000073008.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000073010.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000073016.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000073060.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000073062.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000073130.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000073131.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000073139.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000073176.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000073274.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000074527.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000075359.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000076258.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000078122.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000078314.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000078317.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000078894.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000078897.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079190.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079192.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079222.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079297.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079316.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079317.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079418.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079435.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079480.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079481.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079487.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079508.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079509.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079641.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079794.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079808.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079834.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000080718.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000080774.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000080775.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000080845.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000080877.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000080896.5, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000080923.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000080942.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000080968.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000080974.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081004.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081049.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081137.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081145.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081159.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081187.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081221.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081232.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081235.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081281.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081289.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081350.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081373.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081418.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081603.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081620.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081684.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081718.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081729.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081739.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081752.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082070.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082076.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082109.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082121.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082154.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082189.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082197.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082245.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082260.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082284.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082317.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082319.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082336.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082383.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082423.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082625.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082655.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082675.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082693.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082718.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082729.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082755.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082776.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082778.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082838.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082905.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082926.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082927.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083011.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083019.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083033.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083089.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083199.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083200.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083258.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083283.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083307.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083311.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083327.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083362.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083364.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083367.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083411.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083621.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083646.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083668.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083670.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083736.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083754.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083798.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083823.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083841.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083892.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083906.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083960.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084131.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084166.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084216.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084284.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084289.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084323.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084331.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084348.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084349.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084361.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084616.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084771.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084830.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084932.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000085314.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000085396.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000085587.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000085611.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000085711.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000086049.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000086316.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000086370.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000086404.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000086629.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000086691.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000086695.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000086736.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000086877.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000087159.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000087174.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000087235.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000087252.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000087365.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000087368.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000087396.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000087403.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000087424.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000087569.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000090110.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000090141.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000094054.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000094514.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000094621.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000094728.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000094799.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000095041.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000095092.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000095180.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000095500.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000095595.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000095597.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000095672.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000095742.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000096056.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000096100.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000096768.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000097468.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000097497.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000097571.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000097971.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000098178.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000098559.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000098973.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000099876.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000100104.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000101499.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000101939.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000102426.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000103436.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000103697.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000104028.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000106106.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000109459.1, due to being absent in the computeMatrix output.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools/3.5.0;computeMatrix reference-point -S /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_Vector_LPS12.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_WT_LPS12.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_K197R_LPS12.bw /data/manke/group/rabbani/atac_project2078/atac_pipeline_merged_bam/BigWigs/merged_K197Q_LPS12.bw  -R /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/activegenes.bed -a 2000 -b 2000 --referencePoint TSS -p 20  --missingDataAsZero  --skipZeros  --samplesLabel Vector_LPS12 WT_LPS12 K197R_LPS12 K197Q_LPS12  --blackListFileName /data/repository/organisms/GRCm38_ensembl/akundaje/blacklist.bed  -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS12_onMouseTSS.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Skipping ENSMUSG00000000134.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000000223.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000000355.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000000787.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000000838.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000000881.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000001127.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000001128.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000001131.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000001134.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000001173.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000001924.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000001962.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000001964.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000002010.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000002014.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000002015.5, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000004221.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000004317.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000006373.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000006423.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000006678.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000008035.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000008682.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000009406.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000009731.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000009995.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000015214.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000015217.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000015289.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000015290.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000015291.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000015340.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000015342.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000015668.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000016239.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000016308.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000016319.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000016382.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000016409.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000016427.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000016534.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000017057.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000019087.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000019088.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000019558.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000020457.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000023074.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000023764.18, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025037.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025040.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025058.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025059.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025246.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025257.5, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025260.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025261.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025262.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025264.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025266.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025268.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025269.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025271.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025283.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025287.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025289.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025332.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025525.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025531.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025626.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025630.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025742.5, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025860.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000025862.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031007.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031010.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031012.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031059.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031060.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031065.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031066.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031079.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031093.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031095.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031101.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031103.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031104.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031105.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031107.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031109.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031125.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031133.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031134.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031137.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031138.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031142.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031143.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031144.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031145.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031146.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031147.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031148.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031149.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031150.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031153.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031154.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031155.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031156.18, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031157.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031158.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031161.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031165.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031166.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031167.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031168.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031169.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031171.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031174.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031176.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031196.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031197.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031198.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031200.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031201.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031207.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031214.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031216.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031217.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031220.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031221.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031226.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031227.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031229.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031231.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031232.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031242.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031245.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031246.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031256.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031257.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031258.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031262.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031264.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031266.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031273.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031274.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031278.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031289.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031292.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031295.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031299.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031303.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031304.18, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031309.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031310.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031311.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031312.5, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031314.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031320.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031328.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031333.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031337.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031347.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031349.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031352.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031353.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031357.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031358.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031360.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031365.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031367.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031370.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031371.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031373.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031378.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031381.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031382.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031383.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031385.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031386.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031387.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031388.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031389.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031391.18, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031392.18, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031393.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031398.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031399.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031400.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031402.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031403.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031410.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031422.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031425.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031429.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031431.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031432.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031433.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031434.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000031438.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000032750.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000032806.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000033777.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000033792.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000033983.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000034055.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000034160.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000034248.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000034311.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000034403.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000034457.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000034480.18, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000035062.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000035150.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000035232.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000035299.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000035476.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000035725.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000035776.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000035847.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000035967.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000036022.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000036131.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000036537.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000036572.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000036769.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000036916.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000036932.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000036959.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000036985.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000037217.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000037315.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000037341.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000037347.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000037369.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000037475.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000037636.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000038344.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000039201.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000039231.18, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000039382.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000039545.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000039556.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000040363.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000040522.5, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000040586.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000040621.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000040749.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000040855.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000040990.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000041096.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000041115.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000041133.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000041229.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000041353.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000041552.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000041633.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000041649.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000041658.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000041718.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000042225.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000042271.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000042473.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000042498.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000042595.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000042712.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000042903.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000043384.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000043929.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000044148.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000044149.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000044150.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000044348.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000044583.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000045237.5, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000045427.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000045694.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000046032.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000046269.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000046432.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000046873.18, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000046993.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000047045.17, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000047344.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000047694.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000047757.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000048007.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000048047.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000048592.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000048600.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000048621.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000048949.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000048970.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000049775.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000050029.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000050148.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000050232.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000050332.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000050379.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000050900.5, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000051220.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000051224.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000051412.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000051537.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000052364.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000052676.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000052821.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000054293.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000054626.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000054737.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000055188.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000056537.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000056673.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000057421.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000057457.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000059326.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000059334.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000059461.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000059708.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000060090.16, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000060681.15, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000061273.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000061778.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000062070.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000062783.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000062949.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000063018.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000063663.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000063785.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000063816.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000063897.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064127.13, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064337.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064339.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064340.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064341.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064343.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064344.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064345.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064346.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064351.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064354.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064356.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064357.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064358.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064360.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064363.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064367.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064368.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064370.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064371.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000064372.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000065422.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000065642.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000065947.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000067194.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000067369.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000067567.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000067608.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000067873.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000067924.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000067925.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000068243.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000068457.14, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000069045.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000069049.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000071708.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000071748.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000072844.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000072969.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000073008.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000073010.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000073016.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000073060.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000073062.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000073130.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000073131.12, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000073139.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000073176.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000073274.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000074527.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000075359.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000076258.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000078122.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000078314.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000078317.6, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000078606.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000078894.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000078897.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079190.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079192.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079222.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079297.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079316.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079317.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079418.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079435.9, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079480.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079481.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079487.11, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079508.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079509.10, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079641.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079794.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079808.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000079834.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000080718.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000080774.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000080775.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000080845.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000080877.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000080896.5, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000080923.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000080942.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000080968.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000080974.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081004.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081049.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081137.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081145.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081159.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081187.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081221.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081232.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081235.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081281.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081289.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081350.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081373.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081418.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081603.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081620.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081684.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081718.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081729.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081739.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000081752.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082070.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082076.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082109.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082121.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082154.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082189.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082197.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082245.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082260.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082284.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082317.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082319.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082336.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082383.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082423.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082625.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082655.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082675.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082693.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082718.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082729.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082755.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082776.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082778.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082838.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082905.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082926.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000082927.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083011.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083019.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083033.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083089.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083199.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083200.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083258.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083283.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083307.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083311.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083327.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083362.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083364.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083367.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083411.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083621.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083646.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083668.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083670.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083736.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083754.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083798.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083823.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083841.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083892.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083906.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000083960.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084131.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084166.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084216.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084284.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084289.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084323.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084331.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084348.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084349.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084361.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084616.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084771.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084830.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000084932.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000085314.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000085396.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000085587.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000085611.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000085711.4, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000086049.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000086316.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000086370.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000086404.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000086629.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000086691.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000086695.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000086736.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000086877.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000087159.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000087174.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000087235.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000087252.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000087365.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000087368.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000087396.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000087403.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000087424.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000087569.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000090110.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000090141.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000094054.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000094514.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000094621.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000094728.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000094799.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000095041.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000095092.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000095180.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000095500.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000095595.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000095597.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000095672.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000095742.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000096056.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000096100.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000096768.7, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000097468.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000097497.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000097571.8, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000097971.3, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000098178.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000098559.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000098973.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000099876.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000100104.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000101499.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000101939.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000102426.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000103436.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000103697.1, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000106106.2, due to being absent in the computeMatrix output.\n",
      "Skipping ENSMUSG00000109459.1, due to being absent in the computeMatrix output.\n"
     ]
    }
   ],
   "source": [
    "# computematrix\n",
    "for time in ['LPS0', 'LPS3', 'LPS12']:\n",
    "    bws = \"\"\n",
    "    names = \"\"\n",
    "    for index, cond in enumerate([\"Vector\", \"WT\", \"K197R\", \"K197Q\"]):\n",
    "        bws += os.path.join(bw_path, \"_\".join([\"merged\", cond,time])+\".bw\")+\" \"\n",
    "        names += \"_\".join([cond,time])+ \" \"\n",
    "    cmd = \"module load deeptools/3.5.0;\"\n",
    "    cmd += \"computeMatrix reference-point -S \"+bws\n",
    "    cmd += \" -R \"+os.path.join(rna_path, \"activegenes.bed\")\n",
    "    cmd += \" -a 2000 -b 2000 --referencePoint TSS -p 20  --missingDataAsZero  --skipZeros \"\n",
    "    cmd += \" --samplesLabel \"+ names\n",
    "    cmd += \" --blackListFileName /data/repository/organisms/GRCm38_ensembl/akundaje/blacklist.bed \"\n",
    "    cmd += \" -o \"+os.path.join(output_path, time+\"_onMouseTSS.gz\")\n",
    "    print(cmd)\n",
    "    sp.check_output(cmd, shell = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3ba1e64c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "a5c1137b",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load ultraheatmap/1.3.1; addFeatureToMatrix -m /data/akhtar/group2/rabbani/atac_project2078/plots/LPS0_onMouseTSS.gz -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS0_onMouseTSS_deg.gz -t /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/K197Q_LPS0_vs_WT_LPS0.tsv /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/K197R_LPS0_vs_WT_LPS0.tsv /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/vector_LPS0_vs_WT_LPS0.tsv --featureIdColumn 'GeneID' --featureNames 'log2FoldChange' --referencePoint 'TSS' \n",
      "module load ultraheatmap/1.3.1; addFeatureToMatrix -m /data/akhtar/group2/rabbani/atac_project2078/plots/LPS3_onMouseTSS.gz -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS3_onMouseTSS_deg.gz -t /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/K197Q_LPS3_vs_WT_LPS3.tsv /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/K197R_LPS3_vs_WT_LPS3.tsv /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/vector_LPS3_vs_WT_LPS3.tsv --featureIdColumn 'GeneID' --featureNames 'log2FoldChange' --referencePoint 'TSS' \n",
      "module load ultraheatmap/1.3.1; addFeatureToMatrix -m /data/akhtar/group2/rabbani/atac_project2078/plots/LPS12_onMouseTSS.gz -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS12_onMouseTSS_deg.gz -t /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/K197Q_LPS12_vs_WT_LPS12.tsv /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/K197R_LPS12_vs_WT_LPS12.tsv /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/vector_LPS12_vs_WT_LPS12.tsv --featureIdColumn 'GeneID' --featureNames 'log2FoldChange' --referencePoint 'TSS' \n"
     ]
    }
   ],
   "source": [
    "# ultraheatmap\n",
    "for time in ['LPS0', 'LPS3', 'LPS12']:\n",
    "    cmd = \"module load ultraheatmap/1.3.1; addFeatureToMatrix \"\n",
    "    cmd += \"-m \"+os.path.join(output_path, time+\"_onMouseTSS.gz\")\n",
    "    cmd += \" -o \"+os.path.join(output_path, time+\"_onMouseTSS_deg.gz\")\n",
    "    cmd += \" -t \"+os.path.join(rna_path, \"K197Q_\"+time+\"_vs_WT_\"+time+\".tsv\")+ \" \"+os.path.join(rna_path, \"K197R_\"+time+\"_vs_WT_\"+time+\".tsv\")\n",
    "    cmd += \" \"+os.path.join(rna_path, \"vector_\"+time+\"_vs_WT_\"+time+\".tsv\")\n",
    "    cmd += \" --featureIdColumn 'GeneID' --featureNames 'log2FoldChange' --referencePoint 'TSS' \"\n",
    "    print(cmd)\n",
    "    sp.check_output(cmd, shell = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "6d8eee6c",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools/3.5.0; plotHeatmap  -m /data/akhtar/group2/rabbani/atac_project2078/plots/LPS0_onMouseTSS_deg.gz -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS0_onMouseTSS_deg.png --zMin 0 0 0 0 -1 -1 -1 --zMax 1 1 1 1 1 1 1  --colorMap Greens Greens Greens Greens RdBu_r RdBu_r RdBu_r --samplesLabel vector_atac wt_atac kr_atac kq_atac kq_vs_wt kr_vs_wt vector_vs_wt \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"Greens\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"RdBu_r\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:738: MatplotlibDeprecationWarning:\n",
      "\n",
      "The 'alpha' parameter to Colorbar has no effect because it is overridden by the mappable; it is deprecated since 3.3 and will be removed two minor releases later.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:770: MatplotlibDeprecationWarning:\n",
      "\n",
      "savefig() got unexpected keyword argument \"pdd_inches\" which is no longer supported as of 3.3 and will become an error two minor releases later\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools/3.5.0; plotHeatmap  -m /data/akhtar/group2/rabbani/atac_project2078/plots/LPS3_onMouseTSS_deg.gz -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS3_onMouseTSS_deg.png --zMin 0 0 0 0 -1 -1 -1 --zMax 1 1 1 1 1 1 1  --colorMap Greens Greens Greens Greens RdBu_r RdBu_r RdBu_r --samplesLabel vector_atac wt_atac kr_atac kq_atac kq_vs_wt kr_vs_wt vector_vs_wt \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"Greens\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"RdBu_r\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:738: MatplotlibDeprecationWarning:\n",
      "\n",
      "The 'alpha' parameter to Colorbar has no effect because it is overridden by the mappable; it is deprecated since 3.3 and will be removed two minor releases later.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:770: MatplotlibDeprecationWarning:\n",
      "\n",
      "savefig() got unexpected keyword argument \"pdd_inches\" which is no longer supported as of 3.3 and will become an error two minor releases later\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools/3.5.0; plotHeatmap  -m /data/akhtar/group2/rabbani/atac_project2078/plots/LPS12_onMouseTSS_deg.gz -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS12_onMouseTSS_deg.png --zMin 0 0 0 0 -1 -1 -1 --zMax 1 1 1 1 1 1 1  --colorMap Greens Greens Greens Greens RdBu_r RdBu_r RdBu_r --samplesLabel vector_atac wt_atac kr_atac kq_atac kq_vs_wt kr_vs_wt vector_vs_wt \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"Greens\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"RdBu_r\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:738: MatplotlibDeprecationWarning:\n",
      "\n",
      "The 'alpha' parameter to Colorbar has no effect because it is overridden by the mappable; it is deprecated since 3.3 and will be removed two minor releases later.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:770: MatplotlibDeprecationWarning:\n",
      "\n",
      "savefig() got unexpected keyword argument \"pdd_inches\" which is no longer supported as of 3.3 and will become an error two minor releases later\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# heatmap\n",
    "for time in ['LPS0', 'LPS3', 'LPS12']:\n",
    "    cmd = \"module load deeptools/3.5.0; plotHeatmap \"\n",
    "    cmd += \" -m \"+os.path.join(output_path, time+\"_onMouseTSS_deg.gz\")\n",
    "    cmd += \" -o \"+os.path.join(output_path, time+\"_onMouseTSS_deg.png\")\n",
    "    cmd += \" --zMin 0 0 0 0 -1 -1 -1 --zMax 1 1 1 1 1 1 1 \" \n",
    "    cmd += \" --colorMap Greens Greens Greens Greens RdBu_r RdBu_r RdBu_r --samplesLabel vector_atac wt_atac kr_atac kq_atac kq_vs_wt kr_vs_wt vector_vs_wt \"\n",
    "    print(cmd)\n",
    "    sp.check_output(cmd, shell = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "1da5d423",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools/3.5.0; plotHeatmap  -m /data/akhtar/group2/rabbani/atac_project2078/plots/LPS0_onMouseTSS_deg.gz -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS0_onMouseTSS_deg_k4.png --zMin 0 0 0 0 -1 -1 -1 --zMax 1 1 1 1 1 1 1 --kmeans 4 --clusterUsingSamples 5 6 7  --colorMap Greens Greens Greens Greens RdBu_r RdBu_r RdBu_r --sortUsingSamples 5 6 7 --samplesLabel vector_atac wt_atac kr_atac kq_atac kq_vs_wt kr_vs_wt vector_vs_wt \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"Greens\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"RdBu_r\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:738: MatplotlibDeprecationWarning:\n",
      "\n",
      "The 'alpha' parameter to Colorbar has no effect because it is overridden by the mappable; it is deprecated since 3.3 and will be removed two minor releases later.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:770: MatplotlibDeprecationWarning:\n",
      "\n",
      "savefig() got unexpected keyword argument \"pdd_inches\" which is no longer supported as of 3.3 and will become an error two minor releases later\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools/3.5.0; plotHeatmap  -m /data/akhtar/group2/rabbani/atac_project2078/plots/LPS3_onMouseTSS_deg.gz -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS3_onMouseTSS_deg_k4.png --zMin 0 0 0 0 -1 -1 -1 --zMax 1 1 1 1 1 1 1 --kmeans 4 --clusterUsingSamples 5 6 7  --colorMap Greens Greens Greens Greens RdBu_r RdBu_r RdBu_r --sortUsingSamples 5 6 7 --samplesLabel vector_atac wt_atac kr_atac kq_atac kq_vs_wt kr_vs_wt vector_vs_wt \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"Greens\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"RdBu_r\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:738: MatplotlibDeprecationWarning:\n",
      "\n",
      "The 'alpha' parameter to Colorbar has no effect because it is overridden by the mappable; it is deprecated since 3.3 and will be removed two minor releases later.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:770: MatplotlibDeprecationWarning:\n",
      "\n",
      "savefig() got unexpected keyword argument \"pdd_inches\" which is no longer supported as of 3.3 and will become an error two minor releases later\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools/3.5.0; plotHeatmap  -m /data/akhtar/group2/rabbani/atac_project2078/plots/LPS12_onMouseTSS_deg.gz -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS12_onMouseTSS_deg_k4.png --zMin 0 0 0 0 -1 -1 -1 --zMax 1 1 1 1 1 1 1 --kmeans 4 --clusterUsingSamples 5 6 7  --colorMap Greens Greens Greens Greens RdBu_r RdBu_r RdBu_r --sortUsingSamples 5 6 7 --samplesLabel vector_atac wt_atac kr_atac kq_atac kq_vs_wt kr_vs_wt vector_vs_wt \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"Greens\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"RdBu_r\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:738: MatplotlibDeprecationWarning:\n",
      "\n",
      "The 'alpha' parameter to Colorbar has no effect because it is overridden by the mappable; it is deprecated since 3.3 and will be removed two minor releases later.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:770: MatplotlibDeprecationWarning:\n",
      "\n",
      "savefig() got unexpected keyword argument \"pdd_inches\" which is no longer supported as of 3.3 and will become an error two minor releases later\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# heatmap with k = 4 clustering\n",
    "for time in ['LPS0', 'LPS3', 'LPS12']:\n",
    "    cmd = \"module load deeptools/3.5.0; plotHeatmap \"\n",
    "    cmd += \" -m \"+os.path.join(output_path, time+\"_onMouseTSS_deg.gz\")\n",
    "    cmd += \" -o \"+os.path.join(output_path, time+\"_onMouseTSS_deg_k4.png\")\n",
    "    cmd += \" --zMin 0 0 0 0 -1 -1 -1 --zMax 1 1 1 1 1 1 1 --kmeans 4 --clusterUsingSamples 5 6 7 \" \n",
    "    cmd += \" --colorMap Greens Greens Greens Greens RdBu_r RdBu_r RdBu_r --sortUsingSamples 5 6 7 --samplesLabel vector_atac wt_atac kr_atac kq_atac kq_vs_wt kr_vs_wt vector_vs_wt \"\n",
    "    print(cmd)\n",
    "    sp.check_output(cmd, shell = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "157f9ebf",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools/3.5.0; plotHeatmap  -m /data/akhtar/group2/rabbani/atac_project2078/plots/LPS0_onMouseTSS_deg.gz -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS0_onMouseTSS_deg_sortbydeg.png --zMin 0 0 0 0 -1 -1 -1 --zMax 1 1 1 1 1 1 1  --colorMap Greens Greens Greens Greens RdBu_r RdBu_r RdBu_r --sortUsingSamples 5 6 7 --samplesLabel vector_atac wt_atac kr_atac kq_atac kq_vs_wt kr_vs_wt vector_vs_wt \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"Greens\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"RdBu_r\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:738: MatplotlibDeprecationWarning:\n",
      "\n",
      "The 'alpha' parameter to Colorbar has no effect because it is overridden by the mappable; it is deprecated since 3.3 and will be removed two minor releases later.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:770: MatplotlibDeprecationWarning:\n",
      "\n",
      "savefig() got unexpected keyword argument \"pdd_inches\" which is no longer supported as of 3.3 and will become an error two minor releases later\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools/3.5.0; plotHeatmap  -m /data/akhtar/group2/rabbani/atac_project2078/plots/LPS3_onMouseTSS_deg.gz -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS3_onMouseTSS_deg_sortbydeg.png --zMin 0 0 0 0 -1 -1 -1 --zMax 1 1 1 1 1 1 1  --colorMap Greens Greens Greens Greens RdBu_r RdBu_r RdBu_r --sortUsingSamples 5 6 7 --samplesLabel vector_atac wt_atac kr_atac kq_atac kq_vs_wt kr_vs_wt vector_vs_wt \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"Greens\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"RdBu_r\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:738: MatplotlibDeprecationWarning:\n",
      "\n",
      "The 'alpha' parameter to Colorbar has no effect because it is overridden by the mappable; it is deprecated since 3.3 and will be removed two minor releases later.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:770: MatplotlibDeprecationWarning:\n",
      "\n",
      "savefig() got unexpected keyword argument \"pdd_inches\" which is no longer supported as of 3.3 and will become an error two minor releases later\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools/3.5.0; plotHeatmap  -m /data/akhtar/group2/rabbani/atac_project2078/plots/LPS12_onMouseTSS_deg.gz -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS12_onMouseTSS_deg_sortbydeg.png --zMin 0 0 0 0 -1 -1 -1 --zMax 1 1 1 1 1 1 1  --colorMap Greens Greens Greens Greens RdBu_r RdBu_r RdBu_r --sortUsingSamples 5 6 7 --samplesLabel vector_atac wt_atac kr_atac kq_atac kq_vs_wt kr_vs_wt vector_vs_wt \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"Greens\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"RdBu_r\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/heatmapper_utilities.py:89: UserWarning:\n",
      "\n",
      "Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:738: MatplotlibDeprecationWarning:\n",
      "\n",
      "The 'alpha' parameter to Colorbar has no effect because it is overridden by the mappable; it is deprecated since 3.3 and will be removed two minor releases later.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:770: MatplotlibDeprecationWarning:\n",
      "\n",
      "savefig() got unexpected keyword argument \"pdd_inches\" which is no longer supported as of 3.3 and will become an error two minor releases later\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# heatmap sort by deg\n",
    "for time in ['LPS0', 'LPS3', 'LPS12']:\n",
    "    cmd = \"module load deeptools/3.5.0; plotHeatmap \"\n",
    "    cmd += \" -m \"+os.path.join(output_path, time+\"_onMouseTSS_deg.gz\")\n",
    "    cmd += \" -o \"+os.path.join(output_path, time+\"_onMouseTSS_deg_sortbydeg.png\")\n",
    "    cmd += \" --zMin 0 0 0 0 -1 -1 -1 --zMax 1 1 1 1 1 1 1 \" \n",
    "    cmd += \" --colorMap Greens Greens Greens Greens RdBu_r RdBu_r RdBu_r --sortUsingSamples 5 6 7 --samplesLabel vector_atac wt_atac kr_atac kq_atac kq_vs_wt kr_vs_wt vector_vs_wt \"\n",
    "    print(cmd)\n",
    "    sp.check_output(cmd, shell = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "7f5d3f4b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load ultraheatmap/1.3.1; addFeatureToMatrix -m /data/akhtar/group2/rabbani/atac_project2078/plots/LPS0_onMouseTSS.gz -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS0_onMouseTSS_deg_baseMean.gz -t /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/K197Q_LPS0_vs_WT_LPS0.tsv /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/K197R_LPS0_vs_WT_LPS0.tsv /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/vector_LPS0_vs_WT_LPS0.tsv --featureIdColumn 'GeneID' --featureNames 'baseMean' \n",
      "module load ultraheatmap/1.3.1; addFeatureToMatrix -m /data/akhtar/group2/rabbani/atac_project2078/plots/LPS3_onMouseTSS.gz -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS3_onMouseTSS_deg_baseMean.gz -t /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/K197Q_LPS3_vs_WT_LPS3.tsv /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/K197R_LPS3_vs_WT_LPS3.tsv /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/vector_LPS3_vs_WT_LPS3.tsv --featureIdColumn 'GeneID' --featureNames 'baseMean' \n",
      "module load ultraheatmap/1.3.1; addFeatureToMatrix -m /data/akhtar/group2/rabbani/atac_project2078/plots/LPS12_onMouseTSS.gz -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS12_onMouseTSS_deg_baseMean.gz -t /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/K197Q_LPS12_vs_WT_LPS12.tsv /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/K197R_LPS12_vs_WT_LPS12.tsv /data/akhtar/group2/rabbani/rna_project2129/pairwise_comparison_wo_zeros/vector_LPS12_vs_WT_LPS12.tsv --featureIdColumn 'GeneID' --featureNames 'baseMean' \n"
     ]
    }
   ],
   "source": [
    "# ultraheatmap on basemean\n",
    "for time in ['LPS0', 'LPS3', 'LPS12']:\n",
    "    cmd = \"module load ultraheatmap/1.3.1; addFeatureToMatrix \"\n",
    "    cmd += \"-m \"+os.path.join(output_path, time+\"_onMouseTSS.gz\")\n",
    "    cmd += \" -o \"+os.path.join(output_path, time+\"_onMouseTSS_deg_baseMean.gz\")\n",
    "    cmd += \" -t \"+os.path.join(rna_path, \"K197Q_\"+time+\"_vs_WT_\"+time+\".tsv\")+ \" \"+os.path.join(rna_path, \"K197R_\"+time+\"_vs_WT_\"+time+\".tsv\")\n",
    "    cmd += \" \"+os.path.join(rna_path, \"vector_\"+time+\"_vs_WT_\"+time+\".tsv\")\n",
    "    cmd += \" --featureIdColumn 'GeneID' --featureNames 'baseMean' \"\n",
    "    print(cmd)\n",
    "    sp.check_output(cmd, shell = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "d624edcd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools/3.5.0; plotHeatmap  -m /data/akhtar/group2/rabbani/atac_project2078/plots/LPS0_onMouseTSS_deg_baseMean.gz -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS0_onMouseTSS_deg_baseMean.png --zMin 0 0 0 0 0 0 0 --zMax 1 1 1 1 1000 1000 1000 --whatToShow 'heatmap and colorbar'  --colorMap Greens Greens Greens Greens Blues Blues Blues --samplesLabel vector_atac wt_atac kr_atac kq_atac kq_vs_wt kr_vs_wt vector_vs_wt \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"Greens\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"Blues\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:738: MatplotlibDeprecationWarning:\n",
      "\n",
      "The 'alpha' parameter to Colorbar has no effect because it is overridden by the mappable; it is deprecated since 3.3 and will be removed two minor releases later.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:770: MatplotlibDeprecationWarning:\n",
      "\n",
      "savefig() got unexpected keyword argument \"pdd_inches\" which is no longer supported as of 3.3 and will become an error two minor releases later\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools/3.5.0; plotHeatmap  -m /data/akhtar/group2/rabbani/atac_project2078/plots/LPS3_onMouseTSS_deg_baseMean.gz -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS3_onMouseTSS_deg_baseMean.png --zMin 0 0 0 0 0 0 0 --zMax 1 1 1 1 1000 1000 1000 --whatToShow 'heatmap and colorbar'  --colorMap Greens Greens Greens Greens Blues Blues Blues --samplesLabel vector_atac wt_atac kr_atac kq_atac kq_vs_wt kr_vs_wt vector_vs_wt \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"Greens\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"Blues\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:738: MatplotlibDeprecationWarning:\n",
      "\n",
      "The 'alpha' parameter to Colorbar has no effect because it is overridden by the mappable; it is deprecated since 3.3 and will be removed two minor releases later.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:770: MatplotlibDeprecationWarning:\n",
      "\n",
      "savefig() got unexpected keyword argument \"pdd_inches\" which is no longer supported as of 3.3 and will become an error two minor releases later\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "module load deeptools/3.5.0; plotHeatmap  -m /data/akhtar/group2/rabbani/atac_project2078/plots/LPS12_onMouseTSS_deg_baseMean.gz -o /data/akhtar/group2/rabbani/atac_project2078/plots/LPS12_onMouseTSS_deg_baseMean.png --zMin 0 0 0 0 0 0 0 --zMax 1 1 1 1 1000 1000 1000 --whatToShow 'heatmap and colorbar'  --colorMap Greens Greens Greens Greens Blues Blues Blues --samplesLabel vector_atac wt_atac kr_atac kq_atac kq_vs_wt kr_vs_wt vector_vs_wt \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"Greens\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:481: MatplotlibDeprecationWarning:\n",
      "\n",
      "You are modifying the state of a globally registered colormap. In future versions, you will not be able to modify a registered colormap in-place. To remove this warning, you can make a copy of the colormap first. cmap = copy.copy(mpl.cm.get_cmap(\"Blues\"))\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:738: MatplotlibDeprecationWarning:\n",
      "\n",
      "The 'alpha' parameter to Colorbar has no effect because it is overridden by the mappable; it is deprecated since 3.3 and will be removed two minor releases later.\n",
      "\n",
      "/package/anaconda3/envs/deeptools-3.5.0/lib/python3.7/site-packages/deeptools/plotHeatmap.py:770: MatplotlibDeprecationWarning:\n",
      "\n",
      "savefig() got unexpected keyword argument \"pdd_inches\" which is no longer supported as of 3.3 and will become an error two minor releases later\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# heatmap\n",
    "for time in ['LPS0', 'LPS3', 'LPS12']:\n",
    "    cmd = \"module load deeptools/3.5.0; plotHeatmap \"\n",
    "    cmd += \" -m \"+os.path.join(output_path, time+\"_onMouseTSS_deg_baseMean.gz\")\n",
    "    cmd += \" -o \"+os.path.join(output_path, time+\"_onMouseTSS_deg_baseMean.png\")\n",
    "    cmd += \" --zMin 0 0 0 0 0 0 0 --zMax 1 1 1 1 1000 1000 1000 --whatToShow 'heatmap and colorbar' \" \n",
    "    cmd += \" --colorMap Greens Greens Greens Greens Blues Blues Blues --samplesLabel vector_atac wt_atac kr_atac kq_atac kq_vs_wt kr_vs_wt vector_vs_wt \"\n",
    "    print(cmd)\n",
    "    sp.check_output(cmd, shell = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5cd22a34",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "SSH minimus matplotlib3_py3.8",
   "language": "",
   "name": "rik_ssh_maximus_matplotlib3_py38"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
